---
title: "Integration DNAm and expression - control samples"
author: "MS"
date: "8/19/2020"
output: html_document
---


```{r setup,include=FALSE}
knitr::opts_chunk$set(include = TRUE, root.dir ="/Integration", echo = TRUE)

library("readr")
#devtools::install_github("PF2-pasteur-fr/SARTools", build_opts="--no-resave-data")
library(SARTools)
library("DESeq2")
library(ggplot2)
library(tidyverse)
library(ggthemes)
#library("ReportingTools")
#BiocManager::install("EnhancedVolcano")
library("EnhancedVolcano")
#BiocManager::install('grimbough/biomaRt')
packageVersion('biomaRt')
library(biomaRt)
library(minfi)
#devtools::install_bitbucket("ConesaLab/more")
library(MORE)
library(devtools)
#install_github("achilleasNP/IlluminaHumanMethylationEPICmanifest") 
#install_github("achilleasNP/IlluminaHumanMethylationEPICanno.ilm10b5.hg38")
library(IlluminaHumanMethylationEPICanno.ilm10b5.hg38)
library(IlluminaHumanMethylationEPICmanifest)
#install.packages("data.table")
library(data.table)
#BiocManager::install("ComplexHeatmap")
library("ComplexHeatmap")
library(latticeExtra)
library(lattice)
library("scales")
#install.packages("ggsci")
library("ggsci")
library("gridExtra")
library(dendsort)
#install.packages("colorspace")
library("colorspace")
```

# Load data
```{r load data}
##----Methylation data-------
load("/MethData/analysis/DMresults.RData")
load("/MethData/analysis/mSetSqFlt.RData")
mVals = getM(mSetSqFlt)
betas = getBeta(mSetSqFlt)
rm(mSetSqFlt)

#------RNAseq data-----------
#Day 0 vs 7
D0D7= read.table("RNAseq/30_SARTools_Ctrl/tables/Ctrl7vsCtrl0.complete.txt", header = TRUE)
D0D7 = D0D7[order(D0D7$pvalue),]
sum(D0D7$padj < 0.01, na.rm=TRUE)
sigD0D7 = D0D7[which(D0D7$padj < 0.01),]

sum(sigD0D7$log2FoldChange < 0)#Downregulated
sum(sigD0D7$log2FoldChange > 0) #Upregulated

#Day 0 vs 13
D0D13= read.table("RNAseq/30_SARTools_Ctrl/tables/Ctrl13vsCtrl0.complete.txt", header = TRUE)
D0D13 = D0D13[order(D0D13$pvalue),]
sum(D0D13$padj < 0.01, na.rm=TRUE)
sigD0D13 = D0D13[which(D0D13$padj < 0.01),]

sum(sigD0D13$log2FoldChange < 0)#Downregulated
sum(sigD0D13$log2FoldChange > 0) #Upregulated

#Day 0 vs 20
D0D20= read.table("RNAseq/30_SARTools_Ctrl/tables/Ctrl20vsCtrl0.complete.txt", header = TRUE)
D0D20 = D0D20[order(D0D20$pvalue),]
sum(D0D20$padj < 0.01, na.rm=TRUE)
sigD0D20 = D0D20[which(D0D20$padj < 0.01),]

sum(sigD0D20$log2FoldChange < 0)#Downregulated
sum(sigD0D20$log2FoldChange > 0) #Upregulated

#Day 7 vs 13
D13D7= read.table("RNAseq/30_SARTools_Ctrl/tables/Ctrl7vsCtrl13.complete.txt", header = TRUE)
D13D7 = D13D7[order(D13D7$pvalue),]
sum(D13D7$padj < 0.01, na.rm=TRUE)

#Data for D13 vs D7
#Change FC sign for all genes D7 vs D13
D7D13 <- D13D7 %>% dplyr::select(Id,log2FoldChange, pvalue, padj) %>% mutate(log2FoldChange2 = log2FoldChange * -1) #New column with correct sign for day 13-day 7 
sum(D7D13$padj < 0.01, na.rm=TRUE)
D7D13 = D7D13 %>% dplyr::select(Id,log2FoldChange2, pvalue, padj) 
colnames(D7D13) = c("Id","log2FoldChange", "pvalue", "padj")
#D7D13[which(D7D13$log2FoldChange2 < 0),]
#D7D13[which(D7D13$log2FoldChange2 > 0),]
sigD7D13 = D7D13[which(D7D13$padj < 0.01),]

sum(sigD7D13$log2FoldChange < 0)#Downregulated
sum(sigD7D13$log2FoldChange > 0) #Upregulated

#Day 7 vs 20
D20D7= read.table("RNAseq/30_SARTools_Ctrl/tables/Ctrl7vsCtrl20.complete.txt", header = TRUE)
D20D7 = D20D7[order(D20D7$pvalue),]
sum(D20D7$padj < 0.01, na.rm=TRUE)

#Data for D20 vs D7
#Change FC sign for all genes D7 vs D20
D7D20 <- D20D7 %>% dplyr::select(Id,log2FoldChange, pvalue, padj) %>% mutate(log2FoldChange2 = log2FoldChange * -1) #New column with correct sign for day 20-day 7 
sum(D7D20$padj < 0.01, na.rm=TRUE)
D7D20 = D7D20 %>% dplyr::select(Id,log2FoldChange2, pvalue, padj) 
colnames(D7D20) = c("Id","log2FoldChange", "pvalue", "padj")
#sigD7D20[which(sigD7D20$log2FoldChange2 < 0),]
#sigD7D20[which(sigD7D20$log2FoldChange2 > 0),]
sigD7D20 = D7D20[which(D7D20$padj < 0.01),]

sum(sigD7D20$log2FoldChange < 0)#Downregulated
sum(sigD7D20$log2FoldChange > 0) #Upregulated

#Day 13 vs 20
D13D20= read.table("RNAseq/30_SARTools_Ctrl/tables/Ctrl20vsCtrl13.complete.txt", header = TRUE)
D13D20 = D13D20[order(D13D20$pvalue),]
sum(D13D20$padj < 0.01, na.rm=TRUE)
sigD13D20 = D13D20[which(D13D20$padj < 0.05),]

sum(sigD13D20$log2FoldChange < 0)#Downregulated
sum(sigD13D20$log2FoldChange > 0) #Upregulated

```


```{r normalized counts}

#Make normalized count matrix
targetFile <- "RNAseq/Analysis/design-control.txt"                           # path to the design/target file
rawDir <- "RNAseq/20_featureCounts" 
varInt <- "groups"                                    # factor of interest
condRef <- "Ctrl0"                                      # reference biological condition


# loading target file
target <- loadTargetFile(targetFile=targetFile, varInt=varInt, condRef=condRef, batch=NULL)

# loading counts
counts <- round(loadCountData(target=target, rawDir=rawDir, featuresToRemove=NULL))

#design = as.formula(~ groups)
target$groups = factor(target$groups)
design = as.formula(~ groups)
levels(target$groups)

dds <- DESeqDataSetFromMatrix(countData = counts,
                                   colData = target,
                                   design = design)

dds = DESeq(dds, fitType = "parametric")
resultsNames(dds)
normCounts = counts(dds, normalized = TRUE)
#save(normCounts, file = "normCounts.Match.rData") #For heatmat
normcounts.df = as.data.frame(normCounts)
write.table(normcounts.df, file="normcounts.control.csv", sep="\t")
```

```{r Annotation of ensembl#}
library(biomaRt)

#biomartHuman = useMart("ensembl",dataset="hsapiens_gene_ensembl")
#listFilters(biomartHuman)
#listAttributes(biomartHuman)
#myannot = getBM(attributes = c("external_gene_name", "ensembl_gene_id"),
#                filters = "ensembl_gene_id", values=rownames(counts), 
#                mart=biomartHuman)

#save(myannot, file = "Integration/myannot.RData")

load("Integration/myannot.RData")
```

```{r Volcano plots control}
colnames(myannot)[2] = "Id"

D0D7 = merge(D0D7, myannot, by = "Id")
D0D7 = D0D7[order(D0D7$pvalue),]
#write.csv(D0D7, file="D0D7_results.csv")
sum(D0D7$padj < 0.01, na.rm=TRUE)

annot = D0D7 %>% dplyr::filter(padj<10^-10) %>% arrange(desc(abs(log2FoldChange)))
annotUP = annot %>% dplyr::filter(log2FoldChange > 0)
annotUP = annotUP[1:200,]
annotUP = annotUP[order(annotUP$padj),]

annotDOWN = annot %>% dplyr::filter(log2FoldChange < 0)
annotDOWN = annotDOWN[1:200,]
annotDOWN = annotDOWN[order(annotDOWN$padj),]

#Remove LCRNAs and alternative trascripts
annot = c("ZEB1",	"PRSS23",	"FOXG1",	"NCALD",	"NRG1",	"SMOC1",	"ZEB2",	"CDH11",	"GLIS3",	"ME3",	"ZFHX4",	"FGF9",	"DACH1",	"EYA4",	"WNT5B",	"SOX5",	"ELOVL2",	"DCLK2",	"TRPM8",	"DGKI",	"PRTG",	"LHX2",	"KCNK10",	"KCNC1",	"ADAMTS9",	"AKAP6",	"TBC1D9",	"HHAT",	"RAX",	"PTPRN2",	"PLEKHG1",	"ADAM12",	"SOX6",	"DUSP4",	"SULF1",	"FAM13C",	"MPPED1",	"HPSE2",	"IGFBP3",	"CLVS1",	"LRP2",	"COL2A1",	"MAP6",	"PHYHIPL",	"MEIS1",	"SCD5",	"GABRG3",	"DNAH6",	"GRIA1",	"NR2F2",	"CDH20",	"EPHA4",	"FBLN7",	"CCDC92",	"C6orf118",	"TMEM255A",	"FAM222A",	"NKAIN2",	"ADGRG1",	"GPR85",	"SEZ6L",	"RGS20",	"HMGCLL1",	"ANKRD1",	"MECOM",	"SIX6",	"CDH10",	"NPY2R",	"MEG3",	"FABP7",	"TSHZ2",	"PRDM16",	"FLRT3",	"MARCHF1",	"VAX1",	"BMP3",	"NDP",	"ILDR2",	"PCDH17",	"PAX6",	"GAD2",	"SLC1A4",	"ZBTB16",	"MAPK10",	"C5",	"ULBP1",	"DIRAS2",	"STOML3",	"ACTA2",	"FRRS1L",	"ATP8B4",	"NR2F1",	"CCDC177",	"SLITRK1",	"EBF3",	"F2RL2",	"IGFN1",	"KCNJ13",	"DNAH7",	"FILIP1L",
          
"EPCAM",	"PHC1", "CDH1",	"PRDM14",	"CD9",	"TRDN",	"NANOG",	"SCNN1A",	"MARVELD3",	"CRYBG1",	"LCK",	"FLT1",	"LAMC2",	"ESRP2",	"AP1M2",	"IRF6",	"RUNX1T1",	"CCDC141",	"BCL11A",	"EPHA1",	"FAM83F",	"CDCP1",	"GRID2",	"GRHL2",	"SYT2",	"BRINP1",	"NKX1-2",	"HTR7",	"CALB1",	"TDGF1",	"TTN",	"GAL",	"CCDC69",	"GLB1L3",	"RBM47",	"NANOGP8",	"RAB17",	"C13orf42",	"CARD11",	"CHST4",	"POU5F1",	"SCGB3A2",	"MTARC2",	"CTCFL",	"ESRP1",	"A2ML1",	"PMEPA1",	"LAD1",	"FOXD3-AS1",	"CLDN7",	"ABHD12B",	"PPIC",	"TRIML2",	"CRB3",	"HPAT5",	"HHLA1",	"SYK",	"FGF19",	"IDO1",	"SLC17A9",	"RARRES2",	"GABRP",	"SH2D3A",	"C1orf210",	"FOXD3",	"MGAT4C",	"FOXH1",	"GDF3",	"TNFRSF8",	"USP43",	"OR52A1",	"BICDL2",	"ZNF488",	"LAMC3",	"TMEM30B",	"CCDC172",	"SHISA3",	"ITLN2",	"MYCT1",	"TM4SF18",	"CCL26",	"POU3F4",	"SLC15A1",	"RIT2",	"TACR1",	"BCL2L14",	"KLK13",	"AADACL3",	"SEMG1",	"TMC1",	"TINAGL1",	"PRR15",	"BSPRY",	"OVOL2",	"UCMA",	"VAV1",	"LEFTY2",	"SLC51A",	"FGF4",	"KCTD8",	"TFAP2C",	"IL20RA")
  

keyvals.colour <- ifelse(
    D0D7$padj >= 0.01 |  D0D7$padj == "NA", 'grey30',
      ifelse(D0D7$padj < 0.01, 'red2',
        'grey30'))
  keyvals.colour[is.na(keyvals.colour)] <- 'grey30'
  names(keyvals.colour)[keyvals.colour == 'red2'] <- 'Significant'
  names(keyvals.colour)[keyvals.colour == 'black'] <- '?'
  names(keyvals.colour)[keyvals.colour == 'grey30'] <- 'Not significant'

pdf("CtrlDay0vsDay7.EnhVolA.pdf", width = 12, height = 12)
EnhancedVolcano(D0D7,
    lab = D0D7$external_gene_name,
    selectLab =annot,
    labSize = 2,
    x = 'log2FoldChange',
    y = 'pvalue',
    hline = NULL,
    FCcutoff = 0.5,
    cutoffLineType = "blank",
    colCustom = keyvals.colour,
    title = "Day 0 vs day 7",
    caption = "")
dev.off()


##DAY 0 vs 13 -------------------------------------------------------
sum(D0D13$padj < 0.01, na.rm=TRUE)

D0D13 = merge(D0D13, myannot, by = "Id")
D0D13 = D0D13[order(D0D13$pvalue),]
#write.csv(D0D13, file="D0D13_results.csv")


annot = D0D13%>% dplyr::filter(padj<10^-10) %>% arrange(desc(abs(log2FoldChange)))
annotUP = annot %>% dplyr::filter(log2FoldChange > 0)
annotUP = annotUP[1:200,]
annotUP = annotUP[order(annotUP$padj),]

annotDOWN = annot %>% dplyr::filter(log2FoldChange < 0)
annotDOWN = annotDOWN[1:200,]
annotDOWN = annotDOWN[order(annotDOWN$padj),]


#Remove LCRNAs and alternative trascripts
annot = c("ZEB1",	"FOXG1",	"PCDH19",	"DCLK2",	"EYA4",	"NPAS3",	"POU3F3",	"MAP6",	"SMOC1",	"TBC1D9",	"SOX6",	"PRDM16",	"ZEB2",	"PTPRN2",	"WNT5B",	"MPPED1",	"TMEM255A",	"FGF9",	"PHYHIPL",	"SOX5",	"CNR1",	"LHX2",	"GRIA1",	"FAM13C",	"KCNC1",	"MYO16",	"MEIS1",	"NOG",	"VAX1",	"RSPO1",	"PCDH9",	"GPR85",	"ANKRD1",	"CCDC92",	"ISLR2",	"COL2A1",	"AMER2",	"PCDH17",	"CDH10",	"DLX6-AS1",	"MECOM",	"SCD5",	"ST8SIA4",	"CDH20",	"MMRN1",	"ADGRG1",	"GPR50",	"LRP2",	"OTX2-AS1",	"DIO3",	"ZIC4",	"DLX1",	"ZBTB16",	"NHLH1",	"PRRX1",	"FBLN7",	"NKX2-1",	"DLX6",	"ISL1",	"FLRT3",	"SCN3A",	"RTL1",	"CRB1",	"LGR5",	"MEG3",	"EMX2OS",	"NR2F1",	"VCAM1",	"SOX1",	"GAD2",	"SFTA3",	"UNC13C",	"CHL1",	"DCT",	"CNTN6",	"ST18",	"OLIG2",	"NR2F2",	"F2RL2",	"FILIP1L",	"STMN4",	"CELF3",	"EMX2",	"FOXG1-AS1",	"LHX9",	"HES5",	"ASCL1",	"HDC",	"SLC30A4",	"DLX2",	"GASK1B",	"DMTN",	"DLX5",	"ZIC1",	"TLR4",	"NR2E1",	"SCRG1",		"C21orf62","FEZF2",	"FGF8",		"PAX6",	"PLCH2","NDST4",	"ONECUT3",	"SHH",	"HCRT",	"NEUROD1",	"NNMT",	"WNT7B",	"DLK1",
          
"CDH1",	"NANOG",	"CNMD",	"FLT1",	"VRTN",	"TRDN",	"CRYBG1",	"ADGRG2",	"AP1M2",	"CXCL5",	"TC2N",	"GRID2",	"FAM160A1",	"RAB17",	"MARVELD3",	"GAL",	"ESRP1",	"SPINT1",	"RBM47",	"LRRC2",	"IRF6",	"B3GNT7",	"HTR7",	"NTS",	"TRIML2",	"LRRK1",	"TDGF1",	"POU5F1",	"ESRP2",	"NANOGP8",	"CHST4",	"ZFP42",	"ZSCAN10",	"FGD5",	"KCNK6",	"CLDN7",	"SCGB3A2",	"CTCFL",	"GRB7",	"CARD11",	"NKX1-2",	"SCNN1A",	"TUBA4A",	"ABHD12B",	"DHRS3",	"VIL1",	"LCK",	"ARHGEF35",	"PRDM14",	"GRHL2",	"NOS1",	"CDCP1",	"FOXD3-AS1",	"EPHA1",	"IDO1",	"COL6A3",	"HHLA1",	"HPAT5",	"PRSS8",	"GABRP",	"C13orf42",	"KLRG2",	"NR5A2",	"TMEM92",	"FOXL2NB",	"SYK",	"LAD1",	"AIF1",	"CRB3",	"A2ML1",	"GRM4",	"TNFSF11",	"ACTN3",	"TMEM30B",	"FUT2",	"FOXD3",	"FOXH1",	"GDF3",	"ST14",	"DAZL",	"SH2D3A",	"MLPH",	"C1orf210",	"KDF1",	"OR52A1",	"C3",	"CCDC172",	"ACP5",	"TMEM125",	"CCL26",	"TINAGL1",	"RIT2",	"LTBR",	"LY75",	"FAM124B",	"TNFRSF8",	"FGF4",	"BSPRY",	"C1orf116",	"PRR15",

"PRSS23",	"EPCAM",	"PODXL",	"MAP7",	"DCLK2",	"NCALD")

#Tatt ut opp: 	"NKX2-1-AS1",	"FAM71F2",	"AMBN",	, "ANKFN1",	"DNAH10OS",	"DIO3OS"	"GUCY1A1",	"TRIM67",	"TKTL1",	"SLC40A1",

keyvals.colour <- ifelse(
    D0D13$padj >= 0.01 |  D0D13$padj == "NA", 'grey30',
      ifelse(D0D13$padj < 0.01, 'red2',
        'grey30'))
  keyvals.colour[is.na(keyvals.colour)] <- 'grey30'
  names(keyvals.colour)[keyvals.colour == 'red2'] <- 'Significant'
  names(keyvals.colour)[keyvals.colour == 'black'] <- '?'
  names(keyvals.colour)[keyvals.colour == 'grey30'] <- 'Not significant'
  
pdf("CtrlDay0vsDay13.EnhVolA.pdf", width = 12, height = 12)
EnhancedVolcano(D0D13,
    lab = D0D13$external_gene_name,
    selectLab = annot,
    labSize = 2.0,
    x = 'log2FoldChange',
    y = 'pvalue',
    FCcutoff = 0.5,
    cutoffLineType = "blank",
    colCustom = keyvals.colour,
    title = "Day 0 vs day 13",
    caption = "")
dev.off()


### DAY 0 vs 20 -------------------------------------------------------
sum(D0D20$padj < 0.01, na.rm=TRUE)
D0D20 = merge(D0D20, myannot, by = "Id")
D0D20 = D0D20[order(D0D20$pvalue),]
#write.csv(D0D20, file="D0D20_results.csv")

annot = D0D20%>% dplyr::filter(padj<10^-10) %>% arrange(desc(abs(log2FoldChange)))
annotUP = annot %>% dplyr::filter(log2FoldChange > 0)
annotUP = annotUP[1:200,]
annotUP = annotUP[order(annotUP$padj),]

annotDOWN = annot %>% dplyr::filter(log2FoldChange < 0)
annotDOWN = annotDOWN[1:200,]
annotDOWN = annotDOWN[order(annotDOWN$padj),]


#Remove LCRNAs and alternative trascripts
annot = c("STMN2",	"ZEB1",	"FOXG1",	"CNR1",	"NPAS3",	"MAP6",	"DCLK2",	"POU3F3",	"EYA4",	"SOX6",	"ZEB2",	"ISL1",	"AMER2",	"GPR85",	"RSPO1",	"MYO16",	"NSG2",	"ISLR2",	"ZNF536",	"ZIC4",	"NOG",	"LHX2",	"SCN3A",	"CCDC92",	"VAX1",	"DLX1",	"SAMD5",	"CDH10",	"DLX6-AS1",	"PRRX1",	"OTX2-AS1",	"MPPED1",	"CNTNAP4",	"CELF3",	"SEZ6L",	"CHL1",	"CRB1",	"MMRN1",	"NKX2-1",	"STMN4",	"EMX2OS",	"ONECUT3",	"CDH12",	"INSRR",	"ST18",	"SFTA3",	"ZBTB16",	"DCT",	"LHX9",	"NTRK1",	"ASCL1",	"C21orf62",	"CDK5R2",	"EMX2",	"DLX2",	"GAD2",	"SOX1",	"TRIM67",	"MYT1L",	"DMTN",	"ANKFN1",	"KLHL1",	"GRIA2",	"SLC30A4",	"DDC",	"HDC",	"CNTN6",	"DLX6",	"ZIC1",	"OLIG2",	"FOXG1-AS1",	"NKX2-1-AS1",	"HES5",	"DCC",	"FEZF2",	"CNGA3",	"INSM2",	"LRRTM2",	"NR2F1",	"CNGB1",	"NEUROD1",	"C22orf42",	"MKRN3",	"APC2",	"IL13RA2",	"NDST4",	"GPR22",	"SMIM18",	"IFI44L",	"CYSLTR2",	"ZNF804A",	"NEUROG1",	"CBLN2",	"DLX5",	"GRIN3A",	"ARX",	"ADGRA1",	"FGF8",	"TNR",	"TFAP2B",
          
 "EPCAM",	"CGNL1",	"CDH1",	"NANOG",	"APELA",	"FLT1",	"CRYBG1",	"POLR3G",	"TDGF1",	"RAB17",	"AP1M2",	"EPHA1",	"CXCL5",	"TC2N",	"CYP2S1",	"RBM47",	"NKX1-2",	"LRRK1",	"ESRP1",	"HTR7",	"MARVELD3",	"L1TD1",	"TRDN",	"NTS",	"LRRC2",	"POU5F1",	"ESRP2",	"NANOGP8",	"IRF6",	"SPINT1",	"ZFP42",	"SCNN1A",	"VRTN",	"ABHD12B",	"FOXA3",	"TFCP2L1",	"SCGB3A2",	"GAL",	"DHRS3",	"ARHGEF35",	"CHST4",	"CARD11",	"PRDM14",	"GRB7",	"LCK",	"CLDN7",	"TRIML2",	"GRHL2",	"HHLA1",	"IDO1",	"ZSCAN10",	"CTCFL",	"TUBA4A",	"FOXD3-AS1",	"CRB3",	"TMEM92",	"GABRP",	"PRSS8",	"HPAT5",	"C13orf42",	"LAD1",	"LHX4",	"ADGRE2",	"DAZL",	"ACTN3",	"C1orf210",	"TNFRSF8",	"TMEM30B",	"FUT2",	"TNFSF11",	"FOXD3",	"GDF3",	"A2ML1",	"TNNT2",	"NPR1",	"KDF1",	"LTBR",	"OR52A1",	"C3",	"BCL2L14",	"TMEM125",	"VSIG1",	"CCDC172",	"ACP5",	"DPEP3",	"OVOL2",	"FOXH1",	"KLHDC7A",	"M1AP",	"SLC51A",	"SMIM24",	"LY75",	"EMX1",	"SH2D3A",	"ATP8B1",	"TMPRSS11E",	"COL21A1",	"BSPRY",	"MUC16",	"AIF1",
 
 "RTN1",	"ST8SIA2",	"ZFHX3",	"NCAM1",	"KLHDC8A",	"ASTN1",	"PODXL",	"F11R"
)

#Manuelt tatt ut

keyvals.colour <- ifelse(
    D0D20$padj >= 0.01 |  D0D20$padj == "NA", 'grey30',
      ifelse(D0D20$padj < 0.01, 'red2',
        'grey30'))
  keyvals.colour[is.na(keyvals.colour)] <- 'grey30'
  names(keyvals.colour)[keyvals.colour == 'red2'] <- 'Significant'
  names(keyvals.colour)[keyvals.colour == 'black'] <- '?'
  names(keyvals.colour)[keyvals.colour == 'grey30'] <- 'Not significant'
  
pdf("CtrlDay0vsDay20.EnhVolA.pdf", width = 12, height = 12)
EnhancedVolcano(D0D20,
    lab = D0D20$external_gene_name,
    selectLab = annot,
    x = 'log2FoldChange',
    y = 'pvalue',
    labSize = 2.0,
    FCcutoff = 0.5,
    cutoffLineType = "blank",
    colCustom = keyvals.colour,
    title = "Day 0 vs day 20",
    caption = "")
dev.off()


## Day 7 vs day 13 -------------------------------------
sum(D7D13$padj < 0.01, na.rm=TRUE)

D7D13 = merge(D7D13, myannot, by = "Id")
D7D13 = D7D13[order(D7D13$pvalue),]
#write.csv(D7D13, file="D7D13_results.csv")

annot = D7D13%>% dplyr::filter(padj<10^-10) %>% arrange(desc(abs(log2FoldChange)))
annotUP = annot %>% dplyr::filter(log2FoldChange > 0)
annotUP = annotUP[1:200,]
annotUP = annotUP[order(annotUP$padj),]

annotDOWN = annot %>% dplyr::filter(log2FoldChange < 0)
annotDOWN = annotDOWN[1:200,]
annotDOWN = annotDOWN[order(annotDOWN$padj),]

#Remove LCRNAs and alternative trascripts
annot = c("POU3F3",	"DLX6-AS1",	"PCDH19",	"SLIT2",	"VEPH1",	"NKX2-1",	"TRPS1",	"ANKH",	"CNR1",	"DLX1",	"STMN2",	"ISLR2",	"NOG",	"HSPA4L",	"ABAT",	"ANO1",	"TRIM69",	"LOX",	"NOS2",	"SYT4",	"ST18",	"DLX6",	"CA12",	"EDNRB",	"RUNX1T1",	"OTX2-AS1",	"AMER2",	"LHX9",	"LMCD1",	"SFXN3",	"FGF19",	"MAFB",	"PRRX1",	"CNTN1",	"GRIK3",	"GAS7",	"CHST15",	"CPLX2",	"UNC13C",	"COL24A1",	"GRIK2",	"BCL11A",	"PTCHD1",	"C1orf61",	"MMRN1",	"TGFB3",	"CCNA1",	"SFTA3",	"GAD1",	"EMX2",	"NHLH1",	"DIO3",	"DIO2",	"CCDC69",	"PCSK2",	"SHH",	"PMEPA1",	"FAM135B",	"NFIB",	"SEMA3C",	"OLIG2",	"SAMD5",	"ZNF536",	"HAPLN1",	"EMX2OS",	"NCR3LG1",	"BFSP1",	"NSG2",	"TRPC3",	"FAM181B",	"DMTN",	"INSM2",	"FAM107A",	"INSM1",	"MYORG",	"COL8A1",	"AMBN",	"HSPB8",	"BMP4",	"ZIC3",	"ISL1",	"VWA5B1",	"PRR16",	"GASK1B",	"HEY1",	"C21orf62",	"MAMLD1",	"ONECUT3",	"SLC17A6",	"CNTNAP4",	"SVOP",	"IER3",	"PTPRB",	"LRRTM3",	"PDYN",	"MKRN3",	"NEUROD1",	"ERBB4",	"HCRT",	"SLIT1",

"HS3ST3B1",	"RAX",	"TPBG",	"KCNJ13",	"VRTN",	"HESX1",	"ERBB3",	"UGT3A2",	"MARVELD2",	"POU3F1",	"KCNJ3",	"SH3D19",	"S1PR3",	"DOCK9",	"BMP3",	"NID2",	"LAMP5",	"LMO2",	"ARHGEF5",	"ROBO1",	"PAPLN",	"SLC7A11",	"IGFN1",	"STOML3",	"ZNF788P",	"KCNE3",	"DMKN",	"VLDLR",	"NBAT1",	"F13A1",	"PTN",	"NR6A1",	"FOXA3",	"IRS4",	"MYO5B",	"TRAF5",	"SLC7A7",	"UNC5C",	"FZD7",	"CLMP",	"SHISAL2B",	"PCDH18",	"PAX8",	"MAP7",	"STOM",	"BMP2",	"FAT2",	"APELA",	"TSHZ2",	"SMAGP",	"HLA-DOA",	"ROR2",	"DIRAS2",	"SNTB1",	"PSMB8",	"ROR1",	"PCYT1B",	"PAMR1",	"OPRK1",	"EMX1",	"NRP2",	"NRXN1",	"FERMT1",	"COL14A1",	"ULBP1",	"TMEM108",	"VAMP8",	"MACC1",	"FUT1",	"SEMA3G",	"SFRP2",	"HRK",	"RHOH",	"ADM2",	"RBMS1",	"CNMD",	"ATP8B4",	"PCDH15",	"RASGEF1B",	"ARFGEF3",	"B4GALNT3",	"IGSF11",	"INHBE",	"IFIT1",	"QPCT",	"SPTSSB",	"ADAM12",	"APOL6",	"AURKC",	"ADM",	"PRKCB",	"PRPH2",	"PROKR1",	"CPNE8",	"VWDE",	"CD74",	"LPAR3",	"OPRD1",	"KCNH5",	"HS3ST5"
)

#Manuelt tatt ut

keyvals.colour <- ifelse(
    D7D13$padj >= 0.01 |  D7D13$padj == "NA", 'grey30',
      ifelse(D7D13$padj < 0.01, 'red2',
        'grey30'))
  keyvals.colour[is.na(keyvals.colour)] <- 'grey30'
  names(keyvals.colour)[keyvals.colour == 'red2'] <- 'Significant'
  names(keyvals.colour)[keyvals.colour == 'black'] <- '?'
  names(keyvals.colour)[keyvals.colour == 'grey30'] <- 'Not significant'
  
pdf("CtrlDay7vsDay13.EnhVolA.pdf", width = 12, height = 12)
EnhancedVolcano(D7D13,
    lab = D7D13$external_gene_name,
    selectLab = annot,
    x = 'log2FoldChange',
    y = 'pvalue',
    labSize = 2.0,
    FCcutoff = 0.5,
    cutoffLineType = "blank",
    colCustom = keyvals.colour,
    title = "Day 7 vs day 13",
    caption = "")
dev.off()


## Day 7 vs day 20-------------------------------
sum(D7D20$padj < 0.01, na.rm=TRUE)

D7D20 = merge(D7D20, myannot, by = "Id")
D7D20 = D7D20[order(D7D20$pvalue),]
#write.csv(D7D20, file="D7D20_results.csv")

annot = D7D20%>% dplyr::filter(padj<10^-10) %>% arrange(desc(abs(log2FoldChange)))
annotUP = annot %>% dplyr::filter(log2FoldChange > 0)
annotUP = annotUP[1:200,]
annotUP = annotUP[order(annotUP$padj),]

annotDOWN = annot %>% dplyr::filter(log2FoldChange < 0)
annotDOWN = annotDOWN[1:200,]
annotDOWN = annotDOWN[order(annotDOWN$padj),]

#Remove LCRNAs and alternative trascripts
annot = c("STMN2",	"CNR1",	"POU3F3",	"DLX6-AS1",	"DLX1",	"NFIB",	"SYT4",	"NSG2",	"LHX9",	"AMER2",	"NKX2-1",	"ISLR2",	"BCL11A",	"ZNF536",	"ANO1",	"RUNX1T1",	"CPLX2",	"ST18",	"SFXN3",	"OTX2-AS1",	"C1orf61",	"SAMD5",	"PRRX1",	"CNTNAP4",	"INSM1",	"PCSK2",	"HPCAL4",	"SVOP",	"CHRNB4",	"LRRTM2",	"GAD1",	"ONECUT3",	"PTPRB",	"INSM2",	"MKRN3",	"SPARCL1",	"ZIC4",	"CDH12",	"INSRR",	"SFTA3",	"NTRK1",	"DMTN",	"CCSER1",	"MAMLD1",	"CALB1",	"LRRTM3",	"ISL1",	"C21orf62",	"PMEPA1",	"KLHL1",	"UNC13C",	"FGF19",	"ONECUT1",	"IL13RA2",	"SYT5",	"SLC17A6",	"CDK5R2",	"GRIA2",	"PTPRH",	"NKX2-1-AS1",	"FAM181B",	"GIPR",	"CNGB1",	"OLIG2",	"FAM107A",	"DDC",	"NEUROD1",	"IFI44L",	"AMER3",	"SLC18A1",	"SMPD3",	"FLI1",	"DCT",	"PAX5",	"DLX2",	"ELAVL3",	"TRIM67",	"ARX",	"MLC1",	"MYT1L",	"ASCL1",	"RALYL",	"TFAP2B",	"PDK4",	"CYSLTR2",	"CNGA3",	"NEUROG1",	"HDC",	"NKILA",	"SIAH3",	"VWC2L",	"BRINP2",	"ACSL6",	"USP43",	"SMIM18",	"HEPACAM2",	"TRH",	"OLIG1",	"ACTL6B",	"ZCCHC12",
          
"SFRP1",	"CDH11",	"TPBG",	"RAX",	"FZD7",	"PRSS23",	"HS3ST3B1",	"HESX1",	"NID1",	"CPVL",	"ERBB3",	"NR6A1",	"KCNJ13",	"ROR2",	"IGFBP3",	"SLC7A2",	"RBMS1",	"LRP4",	"LCTL",	"STOM",	"SHISAL2B",	"MARVELD2",	"PMEL",	"PAX8",	"DMKN",	"APELA",	"UGT3A2",	"PRTG",	"CGNL1",	"S1PR3",	"DIAPH2",	"LAMP5",	"SALL1",	"SFRP2",	"JPH1",	"STOML3",	"HPSE2",	"EGF",	"STC2",	"IFIT1",	"PAMR1",	"SIPA1L2",	"ARHGEF5",	"BMP3",	"MAP7",	"EDNRA",	"TGFBR3",	"NR2F2",	"ATF3",	"FAS",	"EYA1",	"ROR1",	"COL14A1",	"DDIT4L",	"VAMP8",	"PCOLCE2",	"IGFN1",	"RASSF3",	"ST8SIA6",	"NBAT1",	"SLC7A11",	"B4GALNT3",	"KCNE3",	"MACC1",	"L1TD1",	"SUSD5",	"FOSB",	"GLI2",	"VRTN",	"FOXA3",	"AOX1",	"ZNF385B",	"NID2",	"ANOS1",	"GAS1",	"MTHFD2",	"MICB",	"DAAM2",	"ADM2",	"ULBP1",	"TRABD2A",	"FAT2",	"FLVCR2",	"PLCG2",	"SNTB1",	"ATP8B4",	"C6orf141",	"SH2D4A",	"CD74",	"EDARADD",	"RASL12",	"NGFR",	"FERMT1",	"SMAGP",	"MYH2",	"MAMDC2",	"PROKR1",	"OPRD1",	"KRT19",	"TCERG1L"
)

keyvals.colour <- ifelse(
    D7D20$padj >= 0.01 |  D7D20$padj == "NA", 'grey30',
      ifelse(D7D20$padj < 0.01, 'red2',
        'grey30'))
  keyvals.colour[is.na(keyvals.colour)] <- 'grey30'
  names(keyvals.colour)[keyvals.colour == 'red2'] <- 'Significant'
  names(keyvals.colour)[keyvals.colour == 'black'] <- '?'
  names(keyvals.colour)[keyvals.colour == 'grey30'] <- 'Not significant'
  
pdf("CtrlDay7vsDay20.EnhVolA.pdf", width = 12, height = 12)
EnhancedVolcano(D7D20,
    lab = D7D20$external_gene_name,
    selectLab = annot,
    x = 'log2FoldChange',
    y = 'pvalue',
    FCcutoff = 0.5,
    labSize = 2.0,
    cutoffLineType = "blank",
    colCustom = keyvals.colour,
    title = "Day 7 vs day 20",
    caption = "")
dev.off()

#Day13 vs day 20 -----------------------------------------
sum(D13D20$padj < 0.01, na.rm=TRUE)

D13D20 = merge(D13D20, myannot, by = "Id")
D13D20 = D13D20[order(D13D20$pvalue),]
#write.csv(D13D20, file="D13D20_results.csv")

annot = D13D20%>% dplyr::filter(padj<10^-10) %>% arrange(desc(abs(log2FoldChange)))
annotUP = annot %>% dplyr::filter(log2FoldChange > 0)
annotUP = annotUP[1:200,]
annotUP = annotUP[order(annotUP$padj),]

annotDOWN = annot %>% dplyr::filter(log2FoldChange < 0)
annotDOWN = annotDOWN[1:200,]
annotDOWN = annotDOWN[order(annotDOWN$padj),]

#Remove LCRNAs and alternative trascripts
annot = c("DNER",	"BCL11A",	"KLHL1",	"LRRTM2",	"STMN2",	"KCNJ3",	"NSG2",	"NMNAT2",	"EPB41L4A",	"ST8SIA3",	"CNGB1",	"FGF14",	"ISL1",	"RELN",	"INA",	"NFIA",	"INSRR",	"IL13RA2",	"C21orf62",	"ONECUT3",	"MKRN3",	"NFIB",	"TMEM196",	"GRIA2",	"DDC",	"REEP1",	"TMEFF2",	"CDK5R2",	"HPCAL4",	"XKR7",	"CNTNAP4",	"NRXN1",	"CNR1",	"SCN2A",	"XKR4",	"PIPOX",	"SHISA6",	"BRINP1",	"DIRAS2",	"AP3B2",	"MYT1L",	"NTRK1",	"MASP1",	"CNGA3",	"RUNDC3B",	"KCNJ9",	"DOK5",	"FSTL5",	"SLC18A1",	"PTPRN",	"POU2F2",	"LHX9",	"BRINP3",	"STMN4",	"CAMK2A",	"CHGB",	"SMIM18",	"SYP",	"ABLIM3",	"VWC2L",	"ZNF536",	"BICDL1",	"NOS1AP",	"TMEM63C",	"C22orf42",	"CDH12",	"SCN3A",	"PROK2",	"PCDH10",	"JHY",	"GLB1L3",	"SLC17A6",	"SVOP",	"SCN3B",	"HEPACAM2",	"GPR22",	"TFAP2B",	"DISP2",	"KCTD16",	"SYN1",	"SLC4A10",	"OPCML",	"AJAP1",	"KCNK9",	"PTPRH",	"TNR",	"RUNDC3A",	"CHRNB4",	"DRD2",	"CALB1",	"PLPPR4",	"SPARCL1",	"MLC1",	"GOLGA7B",	"ONECUT2",	"CACNA1B",	"GRID2",	"PSD2",	"ARHGAP36",	"NWD2",
          
"SFRP1",	"CAPN6",	"CDH11",	"HPSE2",	"CPVL",	"DSP",	"LRP4",	"CMTM7",	"SLC7A2",	"DIAPH2",	"PVT1",	"NID1",	"IGFBP3",	"DDIT4L",	"RGS5",	"NXPH2",	"SIPA1L2",	"ACTC1",	"STXBP6",	"EPHA7",	"FIRRE",	"SORCS1",	"FZD7",	"LAMB1",	"ADAMTS19",	"CGNL1",	"FBN2",	"GLI2",	"LCTL",	"CCDC169",	"EYA1",	"PDPN",	"PRSS23",	"GJA1",	"AMBN",	"SORL1",	"FAM71F1",	"LTBP1",	"SALL4",	"EGF",	"TRIM71",	"SCRG1",	"JPH1",	"COL25A1",	"TEK",	"LYN",	"SLC38A4",	"DAAM2",	"KCNJ13",	"NR2F1",	"PMEL",	"TNFRSF10D",	"TCF7L2",	"MAMDC2",	"DIO3",	"DOCK8",	"PRTG",	"SMCO4",	"TPM1",	"RASSF3",	"SHISAL2B",	"CTDSPL",	"P3H2",	"CCN1",	"PCOLCE2",	"ENPP2",	"LAMP5",	"ABCA5"
)

#Manuelt tatt ut

keyvals.colour <- ifelse(
    D13D20$padj >= 0.01 |  D13D20$padj == "NA", 'grey30',
      ifelse(D13D20$padj < 0.01, 'red2',
        'grey30'))
  keyvals.colour[is.na(keyvals.colour)] <- 'grey30'
  names(keyvals.colour)[keyvals.colour == 'red2'] <- 'Significant'
  names(keyvals.colour)[keyvals.colour == 'black'] <- '?'
  names(keyvals.colour)[keyvals.colour == 'grey30'] <- 'Not significant'

pdf("CtrlDay13vsDay20.EnhVolA.pdf", width = 12, height = 12)
EnhancedVolcano(D13D20,
    lab = D13D20$external_gene_name,
    selectLab = annot,
    x = 'log2FoldChange',
    y = 'pvalue',
    labSize = 2.0,
    FCcutoff = 0.5,
    cutoffLineType = "blank",
    colCustom = keyvals.colour,
    title = "Day 13 vs day 20",
    caption = "")
dev.off()
```

#Create ranked lists for GSEA, RNAseq
```{r Ranked lists for GSEA}
### D0 vs d7
gsea = data.frame(D0D7)
gsea$fcsign <- sign(gsea$log2FoldChange)  #Sign for up/downregulated genes
gsea$pvalue[gsea$pvalue == 0] = 2.225074e-308 #When p-value is 0, tranform to lowest possible p-value in R to avoid inf after log-transformation
gsea$logP=-log10(gsea$pvalue)       #p-value magnitude (low p: high logp)
gsea$metric = gsea$logP/gsea$fcsign           # logP with sign
gsea <- gsea[order(gsea$metric, decreasing = T),] #Order
gsea <- gsea[!is.infinite(gsea$metric),] #remove infinite values
gsea <- gsea[!is.na(gsea$metric),] #Remove NAs
gsea$ENSID = gsea$Id
gsea = gsea[,c("ENSID","metric")]
rownames(gsea) = NULL

write.table(gsea, sep="\t",file="GSEA-RNA-D0D7.rnk", quote=F, row.names=F, col.names = F)

### D0 vs d13
gsea = data.frame(D0D13)
gsea$fcsign <- sign(gsea$log2FoldChange)  #Sign for up/downregulated genes
gsea$pvalue[gsea$pvalue == 0] = 2.225074e-308 #When p-value is 0, tranform to lowest possible p-value in R to avoid inf after log-transformation
gsea$logP=-log10(gsea$pvalue)   #p-value magnitude (low p: high logp)
gsea$metric = gsea$logP/gsea$fcsign      # logP with sign  
gsea <- gsea[order(gsea$metric, decreasing = T),]
gsea <- gsea[!is.infinite(gsea$metric),]#remove infinite values
gsea <- gsea[!is.na(gsea$metric),]#Remove NAs
gsea$ENSID = gsea$Id
gsea = gsea[,c("ENSID","metric")]
rownames(gsea) = NULL

write.table(gsea, sep="\t",file="GSEA-RNA-D0D13.rnk", quote=F, row.names=F, col.names = F)

### D0 vs d20
gsea = data.frame(D0D20)
gsea$fcsign <- sign(gsea$log2FoldChange)  #Sign for up/downregulated genes
gsea$pvalue[gsea$pvalue == 0] = 2.225074e-308 #When p-value is 0, tranform to lowest possible p-value in R to avoid inf after log-transformation
gsea$logP=-log10(gsea$pvalue)  #p-value magnitude (low p: high logp)
gsea$metric = gsea$logP/gsea$fcsign         # logP with sign
gsea <- gsea[order(gsea$metric, decreasing = T),]
gsea <- gsea[!is.infinite(gsea$metric),]#remove infinite values
gsea <- gsea[!is.na(gsea$metric),]#Remove NAs
gsea$ENSID = gsea$Id
gsea = gsea[,c("ENSID","metric")]
rownames(gsea) = NULL

write.table(gsea, sep="\t",file="GSEA-RNA-D0D20.rnk", quote=F, row.names=F, col.names = F)

### D7 vs d13
gsea = data.frame(D7D13)
gsea$fcsign <- sign(gsea$log2FoldChange) #Sign for up/downregulated genes
gsea$pvalue[gsea$pvalue == 0] = 2.225074e-308 #When p-value is 0, tranform to lowest possible p-value in R to avoid inf after log-transformation
gsea$logP=-log10(gsea$pvalue)  #p-value magnitude (low p: high logp)
gsea$metric = gsea$logP/gsea$fcsign        # logP with sign
gsea <- gsea[order(gsea$metric, decreasing = T),]
gsea <- gsea[!is.infinite(gsea$metric),]#remove infinite values
gsea$ENSID = gsea$Id
gsea <- gsea[!is.na(gsea$metric),]#Remove NAs
gsea = gsea[,c("ENSID","metric")]
rownames(gsea) = NULL

write.table(gsea, sep="\t",file="GSEA-RNA-D7D13.rnk", quote=F, row.names=F, col.names = F)

### D7 vs d20
gsea = data.frame(D7D20)
gsea$fcsign <- sign(gsea$log2FoldChange)  #Sign for up/downregulated genes
gsea$pvalue[gsea$pvalue == 0] = 2.225074e-308 #When p-value is 0, tranform to lowest possible p-value in R to avoid inf after log-transformation
gsea$logP=-log10(gsea$pvalue)   #p-value magnitude (low p: high logp)
gsea$metric = gsea$logP/gsea$fcsign           # logP with sign
gsea <- gsea[order(gsea$metric, decreasing = T),]
gsea <- gsea[!is.infinite(gsea$metric),]#remove infinite values
gsea$ENSID = gsea$Id
gsea <- gsea[!is.na(gsea$metric),]#Remove NAs
gsea = gsea[,c("ENSID","metric")]
rownames(gsea) = NULL

write.table(gsea, sep="\t",file="GSEA-RNA-D7D20.rnk", quote=F, row.names=F, col.names = F)

### D13 vs d20
gsea = data.frame(D13D20)
gsea$fcsign <- sign(gsea$log2FoldChange)  #Sign for up/downregulated genes
gsea$pvalue[gsea$pvalue == 0] = 2.225074e-308 #When p-value is 0, tranform to lowest possible p-value in R to avoid inf after log-transformation
gsea$logP=-log10(gsea$pvalue)   #p-value magnitude (low p: high logp)
gsea$metric = gsea$logP/gsea$fcsign          # logP with sign
gsea <- gsea[order(gsea$metric, decreasing = T),]
gsea <- gsea[!is.infinite(gsea$metric),]#remove infinite values
gsea$ENSID = gsea$Id
gsea <- gsea[!is.na(gsea$metric),]#Remove NAs
gsea = gsea[,c("ENSID","metric")]
rownames(gsea) = NULL

write.table(gsea, sep="\t",file="GSEA-RNA-D13D20.rnk", quote=F, row.names=F, col.names = F)
```

##GSEA plots

```{r GSEA plots}
library(colorspace)
#Day 0 vs day 7---------------------------------------------

up = read.table(file = "GSEA/D0D7.GseaPreranked.1629887770850/gsea_report_for_na_pos_1629887770850.tsv", sep = '\t', header = TRUE)

down = read.table(file = "GSEA/D0D7.GseaPreranked.1629887770850/gsea_report_for_na_neg_1629887770850.tsv", sep = '\t', header = TRUE)


GSEAd0d7 = rbind(up,down)
GSEAd0d7 = GSEAd0d7 %>% dplyr::filter(FDR.q.val< 0.05) %>% arrange(desc(abs(NES)))
GSEAd0d7$NES = as.numeric(GSEAd0d7$NES)
write.csv(GSEAd0d7, file="GSEAd0d7.csv")


pdf("D0D7_GSEAPre.pdf",width = 11, height = 10)
ggplot(GSEAd0d7[1:30,], aes(x = NES , y = reorder(NAME,NES), fill = SIZE))+
  geom_bar(stat = "identity") +
  geom_text(aes(label = sprintf("%0.3f", round(FDR.q.val, digits = 3))), size = 3,vjust = 0.5)+
  scale_fill_continuous_sequential(palette = "Purple-Blue",guide = guide_colorbar(reverse = TRUE), limits = c(0,450), breaks = c(20, 100, 200, 300, 400))+
  labs(title = "Day 0 vs day 7",
       subtitle = "Top 30 GO terms", x = "Normalized enrichment score", y = "")+
  theme_minimal()
dev.off()



#Day 0 vs day 13----------------------------------------------
up = read.table(file = "GSEA/D0D13.GseaPreranked.1629888030417/gsea_report_for_na_pos_1629888030417.tsv", sep = '\t', header = TRUE)

down = read.table(file = "GSEA/D0D13.GseaPreranked.1629888030417/gsea_report_for_na_neg_1629888030417.tsv", sep = '\t', header = TRUE)


GSEAd0d13 = rbind(up,down)
GSEAd0d13 = GSEAd0d13 %>% dplyr::filter(FDR.q.val< 0.05) %>% arrange(desc(abs(NES)))
GSEAd0d13$NES = as.numeric(GSEAd0d13$NES)
GSEAd0d13$FDR.q.val = as.numeric(GSEAd0d13$FDR.q.val)
write.csv(GSEAd0d13, file="GSEAd0d13.csv")

pdf("D0D13_GSEAPre.pdf",width = 12, height = 10)
ggplot(GSEAd0d13[1:30,], aes(x = NES , y = reorder(NAME,NES), fill = SIZE))+
  geom_bar(stat = "identity") +
  geom_text(aes(label = sprintf("%0.4f", round(FDR.q.val, digits = 4))), size = 3,vjust = 0.5)+
  scale_fill_continuous_sequential(palette = "Purple-Blue",
                        guide = guide_colorbar(reverse = TRUE), limits = c(0,450), breaks = c(20, 100, 200, 300, 400))+
  labs(title = "Day 0 vs day 13",
       subtitle = "Top 30 GO terms", x = "Normalized enrichment score", y = "")+
  theme_minimal()
dev.off()


#Day 0 vs day 20----------------------------------------------
up = read.table(file = "GSEA/D0D20.GseaPreranked.1629903691253/gsea_report_for_na_pos_1629903691253.tsv", sep = '\t', header = TRUE)

down = read.table(file = "GSEA/D0D20.GseaPreranked.1629903691253/gsea_report_for_na_neg_1629903691253.tsv", sep = '\t', header = TRUE)


GSEAd0d20 = rbind(up,down)
GSEAd0d20 = GSEAd0d20 %>% dplyr::filter(FDR.q.val< 0.05) %>% arrange(desc(abs(NES)))
GSEAd0d20$NES = as.numeric(GSEAd0d20$NES)
write.csv(GSEAd0d20, file="GSEAd0d20.csv")

pdf("D0D20_GSEAPre.pdf",width = 12, height = 10)
ggplot(GSEAd0d20[1:30,], aes(x = NES , y = reorder(NAME,NES), fill = SIZE))+
  geom_bar(stat = "identity") +
  geom_text(aes(label = sprintf("%0.4f", round(FDR.q.val, digits = 4))), size = 3,vjust = 0.5)+
  scale_fill_continuous_sequential(palette = "Purple-Blue",
                        guide = guide_colorbar(reverse = TRUE), limits = c(0,600), breaks = c(20, 100, 200, 300, 400, 500))+
  labs(title = "Day 0 vs day 20",
       subtitle = "Top 30 GO terms", x = "Normalized enrichment score", y = "")+
  theme_minimal()
dev.off()


#Day 7 vs day 13----------------------------------------------
up = read.table(file = "GSEA/D7D13.GseaPreranked.1629892096039/gsea_report_for_na_pos_1629892096039.tsv", sep = '\t', header = TRUE)

down = read.table(file = "GSEA/D7D13.GseaPreranked.1629892096039/gsea_report_for_na_neg_1629892096039.tsv", sep = '\t', header = TRUE)


GSEAd7d13 = rbind(up,down)
GSEAd7d13 = GSEAd7d13 %>% dplyr::filter(FDR.q.val< 0.05) %>% arrange(desc(abs(NES)))
GSEAd7d13$NES = as.numeric(GSEAd7d13$NES)

#Ingen signifikante
GSEAd7d13 = rbind(up,down)
GSEAd7d13 = GSEAd7d13 %>% arrange(FDR.q.val)
GSEAd7d13 = GSEAd7d13[1:30,]
GSEAd7d13$NES = as.numeric(GSEAd7d13$NES)
write.csv(GSEAd7d13, file="GSEAd7d13.csv")

pdf("D7D13_GSEAPre.pdf",width = 12, height = 10)
ggplot(GSEAd7d13, aes(x = NES , y = reorder(NAME,NES), fill = SIZE))+
  geom_bar(stat = "identity") +
  geom_text(aes(label = sprintf("%0.2f", round(FDR.q.val, digits = 2))), size = 3,vjust = 0.5)+
  scale_fill_continuous_sequential(palette = "Purple-Blue",
                        guide = guide_colorbar(reverse = TRUE), limits = c(0,350), breaks = c(15, 100, 200, 300, 400))+
  labs(title = "Day 7 vs day 13",
       subtitle = "Top 30 GO terms", x = "Normalized enrichment score", y = "")+
  theme_minimal()
dev.off()

#Day 7 vs day 20----------------------------------------------
up = read.table(file = "GSEA/D7D20.GseaPreranked.1629892079343/gsea_report_for_na_pos_1629892079343.tsv", sep = '\t', header = TRUE)

down = read.table(file = "GSEA/D7D20.GseaPreranked.1629892079343/gsea_report_for_na_neg_1629892079343.tsv", sep = '\t', header = TRUE)


GSEAd7d20 = rbind(up,down)
GSEAd7d20 = GSEAd7d20 %>% dplyr::filter(FDR.q.val< 0.05) %>% arrange(desc(abs(NES)))
GSEAd7d20$NES = as.numeric(GSEAd7d20$NES)
write.csv(GSEAd7d20, file="GSEAd7d20.csv")

pdf("D7D20_GSEAPre.pdf",width = 12, height = 10)
ggplot(GSEAd7d20[1:30,], aes(x = NES , y = reorder(NAME,NES), fill = SIZE))+
  geom_bar(stat = "identity") +
  geom_text(aes(label = sprintf("%0.4f", round(FDR.q.val, digits = 4))), size = 3,vjust = 0.5)+
  scale_fill_continuous_sequential(palette = "Purple-Blue",
                        guide = guide_colorbar(reverse = TRUE), limits = c(0,650), breaks = c(20, 100, 200, 300, 400, 500, 600))+
  labs(title = "Day 7 vs day 20",
       subtitle = "Top 30 GO terms", x = "Normalized enrichment score", y = "")+
  theme_minimal()
dev.off()

#Day 13 vs day 20----------------------------------------------
up = read.table(file = "GSEA/D13D20.GseaPreranked.1629903714096/gsea_report_for_na_pos_1629903714096.tsv", sep = '\t', header = TRUE)

down = read.table(file = "GSEA/D13D20.GseaPreranked.1629903714096/gsea_report_for_na_neg_1629903714096.tsv", sep = '\t', header = TRUE)


GSEAd13d20 = rbind(up,down)
GSEAd13d20 = GSEAd13d20 %>% dplyr::filter(FDR.q.val< 0.05) %>% arrange(desc(abs(NES)))
GSEAd13d20$NES = as.numeric(GSEAd13d20$NES)
write.csv(GSEAd13d20, file="GSEAd13d20.csv")

pdf("D13D20_GSEAPre.pdf",width = 12, height = 10)
ggplot(GSEAd13d20[1:30,], aes(x = NES , y = reorder(NAME,NES), fill = SIZE))+
  geom_bar(stat = "identity") +
  geom_text(aes(label = sprintf("%0.4f", round(FDR.q.val, digits = 4))), size = 3,vjust = 0.5)+
  scale_fill_continuous_sequential(palette = "Purple-Blue",
                        guide = guide_colorbar(reverse = TRUE), limits = c(0, 820), breaks = c(20, 200, 400, 600, 800))+
  labs(title = "Day 13 vs day 20",
       subtitle = "Top 30 GO terms", x = "Normalized enrichment score", y = "")+
  theme_minimal()
dev.off()
```


# Annotation

```{r get annotation for downstream analysis}
# get the EPIC annotation data
annoEPIC = getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b5.hg38)

#head(annoEPIC)
annoEPIC = as.data.frame(annoEPIC)
annoEPIC$cgid=row.names(annoEPIC)


annoEPICsub <- annoEPIC[match(rownames(mVals),annoEPIC$Name),
                        c(1:4,12:19,22,24, 31,37,49:ncol(annoEPIC))]
                          
                          
```

#Find unique gene annotation for cpgs
```{r}
#Finn unike annoterte cpg-er


all.genes = annoEPICsub[c("cgid", "UCSC_RefGene_Name")] # subset annotated beta matrix 
s = strsplit(all.genes$UCSC_RefGene_Name, split = ";")

uniqueGenes = data.frame(cgid = rep(all.genes$cgid, sapply(s, length)), external_gene_name = unlist(s))
uniqueGenes = uniqueGenes[!duplicated(uniqueGenes[c(names(uniqueGenes))]),] 

```

#Find number of overlapping DM genes with DE genes

```{r overlapping DM genes with DE genes}
colnames(myannot) = c("external_gene_name", "Id")

#D7 ctrl vs day 0 control
#RNA DEG
sigD0D7 = merge(sigD0D7, myannot, by = "Id") #Annotate significant CpGs with external gene names
sigD0D7 = sigD0D7[order(sigD0D7$pvalue),]

#DM CpGs
sig_b_7 = r_b_7 %>% dplyr::select(cgid,logFC) #Subset only fc and cgid
sig_b_7 = merge(sig_b_7, uniqueGenes, by = "cgid") #annotate with external gene names

length(unique(sig_b_7$cgid)) #Number of unique cpgs annotated to genes
length(intersect(sigD0D7$external_gene_name, sig_b_7$external_gene_name)) #number of overlapping DMG with DEG

#-------------------------------------------------------------
#D13 ctrl vs day 0 control
#RNA DEG
sigD0D13 = merge(sigD0D13, myannot, by = "Id") #Annotate with external gene names
sigD0D13 = sigD0D13[order(sigD0D13$pvalue),]

#DM CpGs
sig_b_13 = r_b_13 %>% dplyr::select(cgid,logFC) #Subset only fc and cgid
sig_b_13 = merge(sig_b_13, uniqueGenes, by = "cgid") #annotate with external gene names

length(unique(sig_b_13$cgid)) #Number of unique cpgs annotated to genes
length(intersect(sigD0D13$external_gene_name, sig_b_13$external_gene_name)) #number of overlapping DMG with DEG

#--------------------------------------------------------------
#D20 ctrl vs day 0 control
#RNA DEG
sigD0D20 = merge(sigD0D20, myannot, by = "Id") #Annotate with external gene names
sigD0D20 = sigD0D20[order(sigD0D20$pvalue),]

#DM CpGs
sig_b_20 = r_b_20 %>% dplyr::select(cgid,logFC) #Subset only fc and cgid
sig_b_20 = merge(sig_b_20, uniqueGenes, by = "cgid") #annotate with external gene names

length(unique(sig_b_20$cgid)) #Number of unique cpgs annotated to genes
length(intersect(sigD0D20$external_gene_name, sig_b_20$external_gene_name)) #number of overlapping DMG with DEG

#--------------------------------------------------------------
#D13 ctrl vs day 7 control
#RNA DEG
sigD7D13 = merge(sigD7D13, myannot, by = "Id") #Annotate with external gene names
sigD7D13 = sigD7D13[order(sigD7D13$pvalue),]

#DM CpGs
sig_7_13 = r_7_13 %>% dplyr::select(cgid,logFC) #Subset only fc and cgid
sig_7_13 = merge(sig_7_13, uniqueGenes, by = "cgid") #annotate with external gene names

length(unique(sig_7_13$cgid)) #Number of unique cpgs annotated to genes
length(intersect(sigD7D13$external_gene_name, sig_7_13$external_gene_name)) #number of overlapping DMG with DEG

#-----------------------------------------------------------------
#D20 ctrl vs day 7 control
#RNA DEG
sigD7D20 = merge(sigD7D20, myannot, by = "Id") #Annotate with external gene names
sigD7D20 = sigD7D20[order(sigD7D20$pvalue),]

#DM CpGs
sig_7_20 = r_7_20 %>% dplyr::select(cgid,logFC) #Subset only fc and cgid
sig_7_20 = merge(sig_7_20, uniqueGenes, by = "cgid") #annotate with external gene names

length(unique(sig_7_20$cgid)) #Number of unique cpgs annotated to genes
length(intersect(sigD7D20$external_gene_name, sig_7_20$external_gene_name)) #number of overlapping DMG with DEG

#------------------------------------------------------------------
#D20 ctrl vs day 13 control
#RNA DEG
sigD13D20 = merge(sigD13D20, myannot, by = "Id") #Annotate with external gene names
sigD13D20 = sigD13D20[order(sigD13D20$pvalue),]

#DM CpGs
sig_13_20 = r_13_20 %>% dplyr::select(cgid,logFC) #Subset only fc and cgid
sig_13_20 = merge(sig_13_20, uniqueGenes, by = "cgid") #annotate with external gene names

length(unique(sig_13_20$cgid)) #Number of unique cpgs annotated to genes
length(intersect(sigD13D20$external_gene_name, sig_13_20$external_gene_name)) #number of overlapping DMG with DEG
```


#Overlapping genes Day 7 - 13- 20
```{r}
length(intersect(sigD7D13$external_gene_name, sigD7D20$external_gene_name))
length(intersect(sigD7D13$external_gene_name, sigD13D20$external_gene_name))
length(intersect(sigD7D20$external_gene_name, sigD13D20$external_gene_name))

a = intersect(sigD7D13$external_gene_name, sigD7D20$external_gene_name)
b = intersect(sigD7D13$external_gene_name, sigD13D20$external_gene_name)
c = intersect(sigD7D20$external_gene_name, sigD13D20$external_gene_name)

write.csv(a, "OL_D7D13_D7D20.csv", sep = ";")
write.csv(b, "OL_D7D13_D13D20.csv", sep = ";")
write.csv(c, "OL_D7D20_D13D20.csv", sep = ";")

d = intersect(intersect(sigD7D13$external_gene_name, sigD7D20$external_gene_name), sigD13D20$external_gene_name)
write.csv(d, "OL_D7D13_D7D20_D13D20.csv")

e = setdiff(sigD7D13$external_gene_name, sigD7D20$external_gene_name)
write.csv(f, "difference_D7D13_D7D20.csv")

f = setdiff(sigD13D20$external_gene_name, sigD7D20$external_gene_name)
write.csv(f, "difference_D7D20_D13D20.csv")
```



##HEATMAP

```{r HEATMAP}
#Find top differentially expressed genes
D0D7genes = sigD0D7[1:30,]
D0D7genes = subset(D0D7genes, select = c("Id", "log2FoldChange", "padj"))
D0D7genes = D0D7genes %>% mutate(comparison = "Day0vs7")

D0D13genes = sigD0D13[1:30,]
D0D13genes = subset(D0D13genes, select = c("Id", "log2FoldChange", "padj"))
D0D13genes = D0D13genes %>% mutate(comparison = "Day0vs13")

D0D20genes = sigD0D20[1:30,]
D0D20genes = subset(D0D20genes, select = c("Id", "log2FoldChange", "padj"))
D0D20genes = D0D20genes %>% mutate(comparison = "Day0vs20")

D7D13genes = sigD7D13[1:20,]
D7D13genes = subset(D7D13genes, select = c("Id", "log2FoldChange", "padj"))
D7D13genes = D7D13genes %>% mutate(comparison = "Day7vs13")

D7D20genes = sigD7D20[1:20,]
D7D20genes = subset(D7D20genes, select = c("Id", "log2FoldChange", "padj"))
D7D20genes = D7D20genes %>% mutate(comparison = "Day7vs20")

D13D20genes = sigD13D20[1:20,]
D13D20genes = subset(D13D20genes, select = c("Id", "log2FoldChange", "padj"))
D13D20genes = D13D20genes %>% mutate(comparison = "Day13vs20")


topGenesCtrl = rbind(D0D7genes,D0D13genes, D0D20genes,D7D13genes, D7D20genes,D13D20genes )
topGenesCtrl = topGenesCtrl[!duplicated(topGenesCtrl$Id),]
colnames(topGenesCtrl) = c("ensembl_gene_id", "log2FoldChange", "padj", "comparison")
colnames(myannot) = c("external_gene_name","ensembl_gene_id" )
topGenesCtrl = merge(topGenesCtrl, myannot, by = "ensembl_gene_id")
topGenesCtrl = topGenesCtrl[order(topGenesCtrl$padj),]

topGenesCtrl_name = topGenesCtrl$external_gene_name
#Subset top genes in normcounts dataframe
normcounts.top = normcounts.df.annot %>% dplyr::filter(external_gene_name %in% topGenesCtrl_name)
rownames(normcounts.top) = normcounts.top$external_gene_name
normcounts.top$ensembl_gene_id = NULL
normcounts.top$external_gene_name = NULL
normcounts.top = log2(normcounts.top)
normcounts.top[normcounts.top == -Inf] = 0

annotation_col = data.frame(
    Day = factor(rep(c("0", "7", "13", "20"), c(5,6,4,4))))
annColor = list(Day = c("0" = "#FF9888","7" =  "#B60A1C", "13" = "#C3CE3D", "20" = "#C0D6E4"))

col_dend = dendsort(hclust(dist(t(normcounts.top))), isReverse = FALSE)

p2 = pheatmap(normcounts.top,show_row_dend = FALSE, annotation_col = annotation_col, annotation_colors = annColor, name = "log2(normalized expression)",  cluster_cols = col_dend)
#, col = colorRampPalette(c("navy", "white", "firebrick3")) (10)

pdf("pheatmap_topgenes.pdf", height = 11, width = 8)
p2
dev.off()


#Day 0 vs day 7 ---------------------------------------------------
D0D7top50 = sigD0D7[1:51,] #One does not have external gene name annotation
D0D7top50 = D0D7top50[!D0D7top50$external_gene_name == "",] #Remove 

normcountsD0D7 = normcounts.df.annot %>% dplyr::filter(normcounts.df.annot$external_gene_name %in% D0D7top50$external_gene_name) %>% dplyr::select(contains("D0") | contains("D7") | ensembl_gene_id |external_gene_name) %>% dplyr::select(!ensembl_gene_id) #Select top 50 genes normalized counts
rownames(normcountsD0D7) = normcountsD0D7$external_gene_name #External gene name as rownames
normcountsD0D7 = normcountsD0D7 %>% dplyr::select(!external_gene_name) %>% log2(.) #Convert to log(normcounts)
normcountsD0D7[normcountsD0D7 == -Inf] = 0 #Convert infinite values to 0

annotation_col = data.frame(Day = factor(rep(c("0", "7"), c(5,6)))) #Annotation data frame
annColor = list(Day = c("0" = "#FF9888","7" =  "#B60A1C")) #Color data frame

D0D7top50 = D0D7top50[order(match(D0D7top50$external_gene_name, rownames(normcountsD0D7))), , drop = FALSE] #Order normcounts same as top50
identical(D0D7top50$external_gene_name, rownames(normcountsD0D7)) #Check that they are same order

p1 = rowAnnotation(FoldChange = anno_barplot(D0D7top50$log2FoldChange)) #Fold change plot
p2 = pheatmap(normcountsD0D7,show_row_dend = FALSE, annotation_col = annotation_col, annotation_colors = annColor, name = "log2(normalized expression)") #heatmap

pdf("heatmat_Day0vs7_pheatmap.pdf", height = 9, width = 8)
p1 + p2
dev.off()


#Day 0 vs day 13 ----------------------------------------------------------------------------------

D0D13top50 = sigD0D13[1:52,]
D0D13top50 = D0D13top50[!D0D13top50$external_gene_name == "",] #Remove 

normcountsD0D13 = normcounts.df.annot %>% dplyr::filter(normcounts.df.annot$external_gene_name %in% D0D13top50$external_gene_name) %>% dplyr::select(contains("D0") | contains("D13") | ensembl_gene_id |external_gene_name) %>% dplyr::select(!ensembl_gene_id)
rownames(normcountsD0D13) = normcountsD0D13$external_gene_name
normcountsD0D13 = normcountsD0D13 %>% dplyr::select(!external_gene_name) %>% log2(.)
normcountsD0D13[normcountsD0D13 == -Inf] = 0

col_dend = dendsort(hclust(dist(t(normcountsD0D13))), isReverse = TRUE)

annotation_col = data.frame(Day = factor(rep(c("0", "13"), c(5,4))))
annColor = list(Day = c("0" = "#FF9888","13" =  "#C3CE3D"))

D0D13top50 = D0D13top50[order(match(D0D13top50$external_gene_name, rownames(normcountsD0D13))), , drop = FALSE]
identical(D0D13top50$external_gene_name, rownames(normcountsD0D13)) #Sjekk at det er samme rekkefølge

p1 = rowAnnotation(FoldChange = anno_barplot(D0D13top50$log2FoldChange))
p2 = pheatmap(normcountsD0D13,show_row_dend = FALSE, annotation_col = annotation_col, annotation_colors = annColor, name = "log2(normalized expression)", cluster_cols = col_dend)

pdf("heatmat_Day0vs13_pheatmap.pdf", height = 9, width = 8)
p1 + p2
dev.off()

#Day 0 vs day 20----------------------------------------------------------------------------------
D0D20top50 = sigD0D20[1:51,]
D0D20top50 = D0D20top50[!D0D20top50$external_gene_name == "",] #Remove 

normcountsD0D20 = normcounts.df.annot %>% dplyr::filter(normcounts.df.annot$external_gene_name %in% D0D20top50$external_gene_name) %>% dplyr::select(contains("D0") | contains("D20") | ensembl_gene_id |external_gene_name) %>% dplyr::select(!ensembl_gene_id)
rownames(normcountsD0D20) = normcountsD0D20$external_gene_name
normcountsD0D20 = normcountsD0D20 %>% dplyr::select(!external_gene_name) %>% log2(.)
normcountsD0D20[normcountsD0D20 == -Inf] = 0

col_dend = dendsort(hclust(dist(t(normcountsD0D20))), isReverse = TRUE)

annotation_col = data.frame(Day = factor(rep(c("0", "20"), c(5,4))))
annColor = list(Day = c("0" = "#FF9888","20" =  "#C0D6E4"))

D0D20top50 = D0D20top50[order(match(D0D20top50$external_gene_name, rownames(normcountsD0D20))), , drop = FALSE]
identical(D0D20top50$external_gene_name, rownames(normcountsD0D20)) #Sjekk at det er samme rekkefølge

p1 = rowAnnotation(FoldChange = anno_barplot(D0D20top50$log2FoldChange))
p2 = pheatmap(normcountsD0D20,show_row_dend = FALSE, annotation_col = annotation_col, annotation_colors = annColor, name = "log2(normalized expression)", cluster_cols = col_dend)

pdf("heatmat_Day0vs20_pheatmap.pdf", height = 9, width = 8)
p1 + p2
dev.off()


#Day 7 vs day 13 ----------------------------------------------------------------------------------
D7D13top50 = sigD7D13[1:51,]
D7D13top50 = D7D13top50[!D7D13top50$external_gene_name == "",] #Remove 

normcountsD7D13 = normcounts.df.annot %>% dplyr::filter(normcounts.df.annot$external_gene_name %in% D7D13top50$external_gene_name) %>% dplyr::select(contains("D7") | contains("D13") | ensembl_gene_id |external_gene_name) %>% dplyr::select(!ensembl_gene_id)
rownames(normcountsD7D13) = normcountsD7D13$external_gene_name
normcountsD7D13 = normcountsD7D13 %>%  dplyr::select(!external_gene_name) %>% log2(.)
normcountsD7D13[normcountsD7D13 == -Inf] = 0

col_dend = dendsort(hclust(dist(t(normcountsD7D13))), isReverse = TRUE)

annotation_col = data.frame(Day = factor(rep(c("7", "13"), c(6,4))))
annColor = list(Day = c("7" =  "#B60A1C","13" =  "#C3CE3D"))

D7D13top50 = D7D13top50[order(match(D7D13top50$external_gene_name, rownames(normcountsD7D13))), , drop = FALSE]
identical(D7D13top50$external_gene_name, rownames(normcountsD7D13)) #Sjekk at det er samme rekkefølge

breaksList = seq(0, 20, by = 2)

p1 = rowAnnotation(FoldChange = anno_barplot(D7D13top50$log2FoldChange))
p2 = pheatmap(normcountsD7D13,show_row_dend = FALSE, annotation_col = annotation_col, annotation_colors = annColor, name = "log2(normalized expression)", cluster_cols = col_dend, breaks = breaksList)

pdf("heatmat_Day7vs13_pheatmap.pdf", height = 9, width = 8)
p1 + p2
dev.off()

#Day 7 vs day 20 ----------------------------------------------------------------------------------
D7D20top50 = sigD7D20[1:50,]
D7D20top50 = D7D20top50[!D7D20top50$external_gene_name == "",] #Remove 

normcountsD7D20 = normcounts.df.annot %>% dplyr::filter(normcounts.df.annot$external_gene_name %in% D7D20top50$external_gene_name) %>% dplyr::select(contains("D7") | contains("D20") | ensembl_gene_id |external_gene_name) %>% dplyr::select(!ensembl_gene_id)
rownames(normcountsD7D20) = normcountsD7D20$external_gene_name
normcountsD7D20 = normcountsD7D20 %>% dplyr::select(!external_gene_name) %>% log2(.)
normcountsD7D20[normcountsD7D20 == -Inf] = 0

col_dend = dendsort(hclust(dist(t(normcountsD7D20))), isReverse = TRUE)

annotation_col = data.frame(Day = factor(rep(c("7", "20"), c(6,4))))
annColor = list(Day = c("7" =  "#B60A1C","20" =  "#C0D6E4"))

D7D20top50 = D7D20top50[order(match(D7D20top50$external_gene_name, rownames(normcountsD7D20))), , drop = FALSE]
identical(D7D20top50$external_gene_name, rownames(normcountsD7D20)) #Sjekk at det er samme rekkefølge


p1 = rowAnnotation(FoldChange = anno_barplot(D7D20top50$log2FoldChange))
p2 = pheatmap(normcountsD7D20,show_row_dend = FALSE, annotation_col = annotation_col, annotation_colors = annColor, name = "log2(normalized expression)", cluster_cols = col_dend)

pdf("heatmat_Day7vs20_pheatmap.pdf", height = 9, width = 8)
p1 + p2
dev.off()

# Day 13 vs day 20 ----------------------------------------------------------------------------------
D13D20top50 = sigD13D20[1:50,]
D13D20top50 = D13D20top50[!D13D20top50$external_gene_name == "",] #Remove 

normcountsD13D20 = normcounts.df.annot %>% dplyr::filter(normcounts.df.annot$external_gene_name %in% D13D20top50$external_gene_name) %>% dplyr::select(contains("D13") | contains("D20") | ensembl_gene_id |external_gene_name) %>% dplyr::select(!ensembl_gene_id)
rownames(normcountsD13D20) = normcountsD13D20$external_gene_name
normcountsD13D20 = normcountsD13D20 %>% dplyr::select(!external_gene_name) %>% log2(.)
normcountsD13D20[normcountsD13D20 == -Inf] = 0

col_dend = dendsort(hclust(dist(t(normcountsD13D20))), isReverse = TRUE)

annotation_col = data.frame(Day = factor(rep(c("13", "20"), c(4,4))))
annColor = list(Day = c("13" =  "#C3CE3D","20" =  "#C0D6E4"))

D13D20top50 = D13D20top50[order(match(D13D20top50$external_gene_name, rownames(normcountsD13D20))), , drop = FALSE]
identical(D13D20top50$external_gene_name, rownames(normcountsD13D20)) #Sjekk at det er samme rekkefølge


p1 = rowAnnotation(FoldChange = anno_barplot(D13D20top50$log2FoldChange))
p2 = pheatmap(normcountsD13D20,show_row_dend = FALSE, annotation_col = annotation_col, annotation_colors = annColor, name = "log2(normalized expression)", cluster_cols = col_dend)

pdf("heatmat_Day13vs20_pheatmap.pdf", height = 9, width = 8)
p1 + p2
dev.off()
```


```{r Plot normalized counts for genes of interest}
daycols = c("#FF9888", "#B60A1C", "#C3CE3D","#C0D6E4")

##--Normalized counts---
geneListControl = read.csv("GeneList.csv", header = FALSE, sep = ";") #CSV with external gene names
geneListControl = geneListControl$V1

normcountsGeneList = normcounts.df.annot %>% dplyr::filter(external_gene_name %in% geneListControl) 

normcountsGeneList = normcountsGeneList %>%  pivot_longer(cols = -c(external_gene_name, ensembl_gene_id), names_to = "label", values_to = "NormalizedCounts" )

normcountsGeneList = merge(normcountsGeneList, target, by = "label")
normcountsGeneList$Group = factor(normcountsGeneList$Group, levels = c("Control_day_0","Control_day_7", "Control_day_13", "Control_day_20"))

pdf("GeneList_RNAseq_counts.pdf", width = 20, height = 15)
normcountsGeneList %>% ggplot(., aes(as.factor(Day), NormalizedCounts)) +
  geom_boxplot(aes(fill = as.factor(Day)))+
  scale_fill_manual(values=daycols) +
  geom_point(aes(fill = as.factor(Day)), shape = 21, col = "black" ,size = 0.5) +
  labs(x = "", y = "Counts")+
  guides(fill=guide_legend(title="Day")) +
  facet_wrap(~ external_gene_name, scales = "free_y") +
  scale_y_continuous(limits=c(0,NA))+
  theme_few()+
  theme(legend.position="bottom")
dev.off()

```


#MultiOmics Regulation analysis
This is the goal of the MORE (Multi-Omics REgulation) method: modelling gene expression as a function of experimental variables, such as diseases or treatments, and the potential regulators of a given gene. The idea is to obtain more specific candidate regulators for the biological system under study by applying
regression models, specifically, generalized linear models (GLM).

## Output: GLMresults

> **ResultsPerGene** is a list with as many elements as genes in GeneExpression object.For each gene, there is a list containing the following information:
  **Y** Data frame with the response variable values for that gene (y), the values fitted by the model              (fitted.y), and the residuals of the GLM (residuals).
  **X** Data frame with all the predictors included in the final model.
  **coefficients** Matrix with the estimated coefficients for significant regulators and their p-values.
  **allRegulators** Data frame with all the initial potential regulators in rows and the following                  information in columns: gene, regulator, omic, area (the third optional column in associations),            filter (if the regulator has been filtered out of the model, this column indicates the reason), and         Sig (1 if the regulator is significant and 0 if not). Regarding the filter column, several values           are possible:
      • **MissingValue:** If the regulator has been filtered out of the study because it
                          has missing values.
      • **LowVariation:** If the regulator has been filtered out of the study because
                          it has lower variability than the threshold set by the user in min.variation                                parameter.
      • **ElasticNet:** If the regulator has been excluded of the model by the ElasticNet variable selection                         procedure.
      • **Model:** When the regulator is included in the initial equation model.
      • **omic mcX X:** For instance, TF mc1 R. This notation is related to highly correlated regulators and                         how they are treated to avoid the multicollinearity problem. Following the TF mc1 R                         example, it means that there are two or more transcription factors (TF omic)                                potentially regulating the gene that are highly correlated (in absolute value). In                          such cases, one of them is chosen as the representative and indicated with R. The                           rest of them are labeled with P if positively correlated with the representative and                         with N7 if negatively correlated. An additional row is then added to this table,                            with the regulator TF mc1 R and the filter label being Model, since only this                               representative is considered in the model. When there are several groups of                                 correlated regulators for the same omic, it is indicated with mc1 , mc2 ,etc.
  **significantRegulators** A character vector containing the significant regulators.
**GlobalSummary** List that contains the following elements:
  **GoodnessOfFit** Matrix that collects the model p-value, the final number of degrees of freedom in the           residuals, the R-squared value (which for GLMs is defined as the percentage of deviance explained by         the model) and the Akaike information criterion (AIC) value for all the genes studied.
  **ReguPerGene** Matrix containing, for each omic and gene, the number of initial regulators, the number of         regulators included in the initial model and the number of significant regulators.
  **GenesNOmodel** List of genes for which the final GLM model could not be obtained. There are three               possible reasons for that and they are indicated: “Too many missing values”, “No predictors after           EN” and “GLM error”, where EN refers to the ElasticNet variable selection procedure.
**Arguments** List with the arguments used to generate the model: experimental design
matrix, minimum degrees of freedom in the residuals, significance level, family distribution, variable      selection, etc


## MORE Day 7 vs day 0
```{r more D0D7}
options(stringsAsFactors = FALSE)

# Data for MORE -------------------------------------------------

design = read.table("Integration/design-control.txt", header = TRUE)

# Response variable Y
#Subset DE genes from day7vsDay0 ctrl:
normcounts.DE = normcounts.df[sigD0D7$Id,]

#Subset samples that are present in both RNAseq and methylation
keep = design$Sample_Name[1:16]
normcounts.DE = normcounts.DE[,keep]
str(normcounts.DE)
normcounts.DE = as.matrix(normcounts.DE)

# Predictors X: regulators
mVals = mVals[,keep]
str(mVals)
mVals = as.data.frame(mVals)
pred = list(mVals)
names(pred) = "methylation"
pred$methylation[!is.finite(as.matrix(pred$methylation))] = NA

# Associations gene - potential regulators
uniqueGenes2 = merge(uniqueGenes ,myannot, by= "external_gene_name" )  
uniqueGenes3 = uniqueGenes2[,c(3, 2, 1)] #Change order: Features in responsevariable has to be first, then regulators
assc = list(uniqueGenes3)
names(assc) ="methylation"

# Experimental design matrix
design2 = subset(design, Sample_Name %in% keep, select = c("Sample_Name", "Day", "Concentration.uM"))
rownames(design2) = design2$Sample_Name
design2 = subset(design2, select = "Day")

# Computing GLMs ---------------------------------------------------------

GLMresultsDay7vsDay0 = GetGLM(GeneExpression = normcounts.DE,
                      associations=assc,
                      data.omics=pred,
                      edesign=design2,
                      Res.df = 10, epsilon = 0.00001,
                      alfa=0.05, MT.adjust = "fdr",
                      family = negative.binomial(theta = 10),
                      elasticnet = 0.5, center = TRUE, scale = FALSE,
                      stepwise="two.ways.backward", 
                      interactions.reg = TRUE,
                      min.variation = 0.5,
                      correlation = 0.9,
                      min.obs = 10)

save(GLMresultsDay7vsDay0, file = "Integration/GLMresultsDay7vsDay0.RData")
#load(file ="Integration/GLMresultsDay7vsDay0.RData")

## Exploring results
head(GLMresultsDay7vsDay0$GlobalSummary$GoodnessOfFit) #Model p-vals. degrees of freedom og residuals, R-squared, AIC
head(GLMresultsDay7vsDay0$GlobalSummary$ReguPerGene) #For each omic/gene: number of initial regulators and significat regulators
head(GLMresultsDay7vsDay0$GlobalSummary$GenesNOmodel) #List of genes with no model obtained. #292 genes with no model

colSums(na.omit(GLMresultsDay7vsDay0$GlobalSummary$ReguPerGene)) #16815 significant regulators
##Per gene, PAX6 as example
GLMresultsDay7vsDay0$ResultsPerGene$ENSG00000007372$significantRegulators#"cg01904582"
head(GLMresultsDay7vsDay0$ResultsPerGene$ENSG00000007372$allRegulators) #The allRegulators table shows, for each gene, their regulators, omic, area, the kind of filter applied and the significance of the regulator. In this case (see filter column), in ChIPseq, the regulators are correlated, so a random representative regulator (R) has been chosen. In addition, it is indicated if the correlation with the representative is negative (N) or positive (P). Other possible filters are also indicated as ElasticNet, LowVariation...while Model means that the regulator was included in the model. On the other hand, the Sig column returns 1 if the regulator was significant in the final model and 0 if not.
GLMresultsDay7vsDay0$ResultsPerGene$ENSG00000007372$coefficients #Estimates coefficients of the significant regulators in the final GLM model for a particular gene and their associated p-values. 


GLMresultsDay7vsDay0$arguments$finaldesign
GLMresultsDay7vsDay0$arguments$edesign$Day = factor(GLMresultsDay7vsDay0$arguments$edesign$Day, levels = c("0", "7","13", "20"))

# Plotting significant regulations ----------------------------------------

#Top 10 genes, Day 7 vs 0
top10 = sigD0D7$Id[1:10]
mygene = top10[1] #ZEB1
par(mfrow = c(1, 2))
plotGLM(GLMresultsDay7vsDay0, gene = mygene, regulator = NULL, 
        xlab = "", reguValues = NULL, plotPerOmic = FALSE, order = FALSE,
        gene.col = 1, regu.col = "green4", replicates = TRUE, 
        cont.var = design2$Day)#, cond2plot = design2$Concentration.uM)



# Extracting significant regulations --------------------------------------
SigniRegulations = RegulationPerCondition(GLMresultsDay7vsDay0, betaTest = FALSE)
#head(SigniRegulations)

#This table shows the significant regulators for each gene. The representative column indicates if the regulator was chosen as the random representative of a correlated group of regulators or, otherwise, which regulator was taken as the representative of the group. When no information is provided in this column, it means that the regulator was not part of a correlated group of regulators. Regulators correlated positively with the representative will have the same coefficients (same sign) than the representative, while negatively correlated regulators will have the same coefficients of the representative but with opposite sign.

#The final columns correspond to the regression coefficients of each regulator for each experimental group. In this case, the experimental design matrix (edesign) contained two conditions, so the column Group0 corresponds to the first condition and Group1 corresponds to the second one. These are the conclusions we can draw from the coefficients:
  #• If two experimental groups have the same coefficients, it means that the regulator has the same effect on the gene in both groups.
  #• If one of the coefficients is 0, it means that the regulator has no effect on the gene under this experimental condition.
  #• Experimental groups with different non-zero coefficients indicate that the regulator affects the gene in all these experimental groups but the magnitude of the effect is not the same for all these groups.


length(unique(SigniRegulations$gene)) # 6441 genes with significant regulators
table(SigniRegulations$omic) #16815 significant methylation regulators


```

### Visualize coefficients
```{r}

#GLMtrp200.df = unlist(GLMresults)
Nomodel = GLMresultsDay7vsDay0$GlobalSummary$GenesNOmodel$gene #Gene names with no model

D0D7Coeff = lapply(GLMresultsDay7vsDay0$ResultsPerGene, '[[', 3) #Extract 3rd element of each list (coefficients)

for (i in Nomodel){ #Remove genes with no model from list (clutters future df)
  D0D7Coeff[[i]] = NULL
}

D0D7Coeff.df <- lapply(D0D7Coeff, data.frame, stringsAsFactors = FALSE) #Make each list element into dfs
D0D7Coeff.df2 = bind_rows(D0D7Coeff.df, .id = "ensembl_gene_id") #Convert to df with extra column with ensemb#
D0D7Coeff.df2$cgid = rownames(D0D7Coeff.df2) #Make cgid a comumn
str(D0D7Coeff.df2)


#D0D7Coeff.df2 %>% filter(str_detect(cgid, "methylation")) #The ones names methylation*** are a collection of several CpGs that are highly correlated, and therefore treated as one in the model to 

D0D7Coeff = D0D7Coeff.df2 %>% dplyr::filter(!str_detect(cgid, "Group")) %>% dplyr::filter(!str_detect(cgid, "(Intercept)")) %>% dplyr::filter(!str_detect(cgid, "methylation"))

D0D7Coeff = merge(D0D7Coeff, annoEPIC, by = "cgid")
D0D7Coeff = D0D7Coeff[order(D0D7Coeff$p.value),]
D0D7Coeff = D0D7Coeff[which(D0D7Coeff$p.value < 0.05),]


#Plot modified Manhattan plot
D0D7Coeff$CHR_hg38 = factor(D0D7Coeff$CHR_hg38, levels = c("chr1", "chr2", "chr3", "chr4", "chr5", "chr6", "chr7", "chr8", "chr9", "chr10", "chr11",
                                                     "chr12", "chr13", "chr14", "chr15", "chr16", "chr17","chr18", "chr19", "chr20", "chr21", "chr22","chrX", "chrY"))

ggplot(D0D7Coeff, aes(factor(Start_hg38), coefficient, col = p.value))+
  geom_point(size=1, alpha=1) +
    facet_grid(~ CHR_hg38, scales="free", space="free")+
    labs(title = "Correlation between significant CpGs and DE genes (Day 7 control vs day 0 control)" )+
    xlab("") + 
    ylab("Correlation coefficient") +
    ylim(-1,1)+theme(axis.ticks = element_blank(),axis.text.x = element_blank())
dev.copy(pdf,paste0("manhatten.D0D7.coeff.pos.pdf"),height = 6, width = 7, paper='special')
dev.off()

ggplot(D0D7Coeff, aes(factor(Start_hg38), coefficient, col = p.value))+
  geom_point(size=1, alpha=1) +
    facet_grid(~ Relation_to_Island, scales="free", space="free")+
    labs(title = "Correlation between significant CpGs and DE genes (Day 7 control vs day 0 control)" )+
    xlab("") + 
    ylab("Correlation coefficientt") +
    ylim(-1,1)+theme(axis.ticks = element_blank(),axis.text.x = element_blank())
dev.copy(pdf,paste0("manhatten.D0D7.coeff.reltoisland.pdf"),height = 6, width = 7, paper='special')
dev.off()

all.refGeneGroup = annoEPIC[c("cgid", "UCSC_RefGene_Group")] # subset annotated beta matrix 
s = strsplit(all.refGeneGroup$UCSC_RefGene_Group, split = ";")

uniqueRelToGenes = data.frame(cgid = rep(all.refGeneGroup$cgid, sapply(s, length)), Relation_to_gene = unlist(s))
uniqueRelToGenes = uniqueRelToGenes[!duplicated(uniqueRelToGenes[c(names(uniqueRelToGenes))]),] 

D0D7Coeff2 = merge(D0D7Coeff,uniqueRelToGenes, by = "cgid" )

ggplot(D0D7Coeff2, aes(factor(Start_hg38), coefficient, col = p.value))+
  geom_point(size=1, alpha=1) +
    facet_grid(~ Relation_to_gene, scales="free", space="free")+
    labs(title = "Correlation between significant CpGs and DE genes ((Day 7 control vs day 0 control)" )+
    xlab("") + 
    ylab("Correlation coefficientt") +
    ylim(-1,1)+theme(axis.ticks = element_blank(),axis.text.x = element_blank())
dev.copy(pdf,paste0("manhatten.D0D7.coeff.reltoGene.pdf"),height = 6, width = 7, paper='special')
dev.off()
```



## MORE Day 13 vs day 0 control

```{r more D0D13}
options(stringsAsFactors = FALSE)

# Data for MORE -------------------------------------------------

# Response variable Y
#Subset DE genes from TRp100:
#sigD0D13
normcounts.DE = normcounts.df[sigD0D13$Id, ] 

#Subset samples that are present in both RNAseq and methylation
normcounts.DE = normcounts.DE[,keep]
str(normcounts.DE)
normcounts.DE = as.matrix(normcounts.DE)

# Predictors X: regulators
#str(pred)

# Associations gene - potential regulators
#str(assc)

# Experimental design matrix
#str(design2)

# Computing GLMs ---------------------------------------------------------

GLMresultsD0vsD13 = GetGLM(GeneExpression = normcounts.DE,
                      associations=assc,
                      data.omics=pred,
                      edesign=design2,
                      Res.df = 10, epsilon = 0.00001,
                      alfa=0.05, MT.adjust = "fdr",
                      family = negative.binomial(theta = 10),
                      elasticnet = 0.5, center = TRUE, scale = FALSE,
                      stepwise="two.ways.backward", 
                      interactions.reg = TRUE,
                      min.variation = 0.5,
                      correlation = 0.9,
                      min.obs = 10)

save(GLMresultsD0vsD13, file = "Integration/GLMresultsD0vsD13.RData")
#load(file ="Integration/GLMresultsD0vsD13.RData")

## Exploring results
head(GLMresultsD0vsD13$GlobalSummary$GoodnessOfFit) #Model p-vals. degrees of freedom og residuals, R-squared, AIC
head(GLMresultsD0vsD13$GlobalSummary$ReguPerGene) #For each omic/gene: number of initial regulators and significat regulators
head(GLMresultsD0vsD13$GlobalSummary$GenesNOmodel) #List of genes with no model obtained. #360 genes with no model

GLMresultsD0vsD13$arguments$edesign$Day = factor(GLMresultsD0vsD13$arguments$edesign$Day, levels = c("0", "7","13", "20"))

SigniRegulations = RegulationPerCondition(GLMresultsD0vsD13, betaTest = FALSE)
length(unique(SigniRegulations$gene)) # 6775 genes with significant regulators
table(SigniRegulations$omic) #18950 significant methylation regulators

# Plotting significant regulations ----------------------------------------

#Top 10 genes, SigD0D13
top10 = sigD0D13$Id[1:10]
mygene = top10[1]  
par(mfrow = c(1, 2))
plotGLM(GLMresultsD0vsD13, gene = mygene, regulator = NULL, 
        xlab = "", reguValues = NULL, plotPerOmic = FALSE, order = FALSE,
        gene.col = 1, regu.col = "green4", replicates = TRUE, 
        cont.var = design2$Day)#, cond2plot = design2$Concentration.uM)



```


## MORE Day 20 vs day 0 control

```{r more D0D20}
options(stringsAsFactors = FALSE)

# Data for MORE -------------------------------------------------

# Response variable Y
#Subset DE genes from TRp100:
#sigD0D20
normcounts.DE = normcounts.df[sigD0D20$Id, ] 

#Subset samples that are present in both RNAseq and methylation
normcounts.DE = normcounts.DE[,keep]
str(normcounts.DE)
normcounts.DE = as.matrix(normcounts.DE)

# Predictors X: regulators
#str(pred)

# Associations gene - potential regulators
#str(assc)

# Experimental design matrix
#str(design2)

# Computing GLMs ---------------------------------------------------------

GLMresultsD0vsD20 = GetGLM(GeneExpression = normcounts.DE,
                      associations=assc,
                      data.omics=pred,
                      edesign=design2,
                      Res.df = 10, epsilon = 0.00001,
                      alfa=0.05, MT.adjust = "fdr",
                      family = negative.binomial(theta = 10),
                      elasticnet = 0.5, center = TRUE, scale = FALSE,
                      stepwise="two.ways.backward", 
                      interactions.reg = TRUE,
                      min.variation = 0.5,
                      correlation = 0.9,
                      min.obs = 10)

save(GLMresultsD0vsD20, file = "Integration/GLMresultsD0vsD20.RData")
#load(file ="Integration/GLMresultsD0vsD20.RData")

## Exploring results
head(GLMresultsD0vsD20$GlobalSummary$GoodnessOfFit) #Model p-vals. degrees of freedom og residuals, R-squared, AIC
head(GLMresultsD0vsD20$GlobalSummary$ReguPerGene) #For each omic/gene: number of initial regulators and significat regulators
head(GLMresultsD0vsD20$GlobalSummary$GenesNOmodel) #List of genes with no model obtained. #12 genes with no model


GLMresultsD0vsD20$arguments$edesign$Day = factor(GLMresultsD0vsD20$arguments$edesign$Day, levels = c("0", "7","13", "20"))

SigniRegulations = RegulationPerCondition(GLMresultsD0vsD20, betaTest = FALSE)
length(unique(SigniRegulations$gene)) # 9071 genes with significant regulators
table(SigniRegulations$omic) #23166 significant methylation regulators


# Plotting significant regulations ----------------------------------------

#Top 10 genes, TRP100
top10 = sigD0D20$Id[1:10]
mygene = top10[1]  
par(mfrow = c(1, 2))
plotGLM(GLMresultsD0vsD20, gene = mygene, regulator = NULL, 
        xlab = "", reguValues = NULL, plotPerOmic = FALSE, order = FALSE,
        gene.col = 1, regu.col = "green4", replicates = TRUE, 
        cont.var = design2$Day)#, cond2plot = design2$Concentration.uM)



GLMresultsD0vsD20$ResultsPerGene$ENSG00000204531
GLMresultsD0vsD20$ResultsPerGene$ENSG00000111704 
GLMresultsD0vsD20$ResultsPerGene$ENSG00000248329
GLMresultsD0vsD20$ResultsPerGene$ENSG00000039068
GLMresultsD0vsD20$ResultsPerGene$ENSG00000149294
GLMresultsD0vsD20$ResultsPerGene$ENSG00000172020
GLMresultsD0vsD20$ResultsPerGene$ENSG00000129757
GLMresultsD0vsD20$ResultsPerGene$ENSG00000077279

GLMresultsD0vsD20$ResultsPerGene$ENSG00000088305
GLMresultsD0vsD20$ResultsPerGene$ENSG00000121570
GLMresultsD0vsD20$ResultsPerGene$ENSG00000065361
GLMresultsD0vsD20$ResultsPerGene$ENSG00000160973
GLMresultsD0vsD20$ResultsPerGene$ENSG00000069482
GLMresultsD0vsD20$ResultsPerGene$ENSG00000148200
GLMresultsD0vsD20$ResultsPerGene$ENSG00000154096
GLMresultsD0vsD20$ResultsPerGene$ENSG00000170558
GLMresultsD0vsD20$ResultsPerGene$ENSG00000080166
GLMresultsD0vsD20$ResultsPerGene$ENSG00000127920
GLMresultsD0vsD20$ResultsPerGene$ENSG00000155980
GLMresultsD0vsD20$ResultsPerGene$ENSG00000110693
GLMresultsD0vsD20$ResultsPerGene$ENSG00000144834
```


## MORE Day 13 vs day 7 control

```{r more D7D13}
options(stringsAsFactors = FALSE)

# Data for MORE -------------------------------------------------

# Response variable Y
#Subset DE genes from TRp100:
#sigD7D13
normcounts.DE = normcounts.df[sigD7D13$Id, ] 

#Subset samples that are present in both RNAseq and methylation
normcounts.DE = normcounts.DE[,keep]
str(normcounts.DE)
normcounts.DE = as.matrix(normcounts.DE)

# Predictors X: regulators
#str(pred)

# Associations gene - potential regulators
#str(assc)

# Experimental design matrix
#str(design2)

# Computing GLMs ---------------------------------------------------------

GLMresultsD7vsD13 = GetGLM(GeneExpression = normcounts.DE,
                      associations=assc,
                      data.omics=pred,
                      edesign=design2,
                      Res.df = 10, epsilon = 0.00001,
                      alfa=0.05, MT.adjust = "fdr",
                      family = negative.binomial(theta = 10),
                      elasticnet = 0.5, center = TRUE, scale = FALSE,
                      stepwise="two.ways.backward", 
                      interactions.reg = TRUE,
                      min.variation = 0.5,
                      correlation = 0.9,
                      min.obs = 10)

save(GLMresultsD7vsD13, file = "Integration/GLMresultsD7vsD13.RData")

#load(file ="Integration/GLMresultsD7vsD13.RData")

## Exploring results
head(GLMresultsD7vsD13$GlobalSummary$GoodnessOfFit) #Model p-vals. degrees of freedom og residuals, R-squared, AIC
head(GLMresultsD7vsD13$GlobalSummary$ReguPerGene) #For each omic/gene: number of initial regulators and significat regulators
head(GLMresultsD7vsD13$GlobalSummary$GenesNOmodel) #List of genes with no model obtained. #3 genes with no model

GLMresultsD7vsD13$arguments$edesign$Day = factor(GLMresultsD7vsD13$arguments$edesign$Day, levels = c("0", "7","13", "20"))

SigniRegulations = RegulationPerCondition(GLMresultsD7vsD13, betaTest = FALSE)
length(unique(SigniRegulations$gene)) #2683 genes with significant regulators
table(SigniRegulations$omic) #8072 significant methylation regulators

# Plotting significant regulations ----------------------------------------

#Top 10 genes, 
top10 = sigD7D13$Id[1:10]
mygene = top10[1]  
#par(mfrow = c(4, 3))
plotGLM(GLMresultsD7vsD13, gene = mygene, regulator = NULL, 
        xlab = "", reguValues = NULL, plotPerOmic = FALSE, order = FALSE,
        gene.col = 1, regu.col = "green4", replicates = TRUE, 
        cont.var = design2$Day)#, cond2plot = design2$Concentration.uM)



```

## MORE Day 20 vs day 7 control

```{r more D7D20}
options(stringsAsFactors = FALSE)

# Data for MORE -------------------------------------------------

# Response variable Y
#Subset DE genes from TRp100:
#sigD7D20
normcounts.DE = normcounts.df[sigD7D20$Id, ] 

#Subset samples that are present in both RNAseq and methylation
normcounts.DE = normcounts.DE[,keep]
str(normcounts.DE)
normcounts.DE = as.matrix(normcounts.DE)

# Predictors X: regulators
#str(pred)

# Associations gene - potential regulators
#str(assc)

# Experimental design matrix
#str(design2)

# Computing GLMs ---------------------------------------------------------

GLMresultsD7vsD20 = GetGLM(GeneExpression = normcounts.DE,
                      associations=assc,
                      data.omics=pred,
                      edesign=design2,
                      Res.df = 10, epsilon = 0.00001,
                      alfa=0.05, MT.adjust = "fdr",
                      family = negative.binomial(theta = 10),
                      elasticnet = 0.5, center = TRUE, scale = FALSE,
                      stepwise="two.ways.backward", 
                      interactions.reg = TRUE,
                      min.variation = 0.5,
                      correlation = 0.9,
                      min.obs = 10)

save(GLMresultsD7vsD20, file = "Integration/GLMresultsD7vsD20.RData")
#load(file ="Integration/GLMresultsD7vsD20.RData")

## Exploring results
head(GLMresultsD7vsD20$GlobalSummary$GoodnessOfFit) #Model p-vals. degrees of freedom og residuals, R-squared, AIC
head(GLMresultsD7vsD20$GlobalSummary$ReguPerGene) #For each omic/gene: number of initial regulators and significat regulators
length(GLMresultsD7vsD20$GlobalSummary$GenesNOmodel) #List of genes with no model obtained. #2 genes with no model

GLMresultsD7vsD20$arguments$edesign$Day = factor(GLMresultsD7vsD20$arguments$edesign$Day, levels = c("0", "7","13", "20"))

SigniRegulations = RegulationPerCondition(GLMresultsD7vsD20, betaTest = FALSE)
length(unique(SigniRegulations$gene)) # 5230 genes with significant regulators
table(SigniRegulations$omic) #14542 significant methylation regulators

# Plotting significant regulations ----------------------------------------

#Top 10 genes, TRP100
top10 = sigD7D20$Id[1:10]
mygene = top10[1]  
par(mfrow = c(1, 2))
plotGLM(GLMresultsD7vsD20, gene = mygene, regulator = NULL, 
        xlab = "", reguValues = NULL, plotPerOmic = FALSE, order = FALSE,
        gene.col = 1, regu.col = "green4", replicates = TRUE, 
        cont.var = design2$Day)#, cond2plot = design2$Concentration.uM)



```

## MORE Day 20 vs day 13 control
```{r more D13D20}
options(stringsAsFactors = FALSE)

# Data for MORE -------------------------------------------------

# Response variable Y
#Subset DE genes from TRp100:
#sigD13D20
normcounts.DE = normcounts.df[sigD13D20$Id, ] 

#Subset samples that are present in both RNAseq and methylation
normcounts.DE = normcounts.DE[,keep]
str(normcounts.DE)
normcounts.DE = as.matrix(normcounts.DE)

# Predictors X: regulators
#str(pred)

# Associations gene - potential regulators
#str(assc)

# Experimental design matrix
#str(design2)

# Computing GLMs ---------------------------------------------------------

GLMresultsD13vsD20 = GetGLM(GeneExpression = normcounts.DE,
                      associations=assc,
                      data.omics=pred,
                      edesign=design2,
                      Res.df = 10, epsilon = 0.00001,
                      alfa=0.05, MT.adjust = "fdr",
                      family = negative.binomial(theta = 10),
                      elasticnet = 0.5, center = TRUE, scale = FALSE,
                      stepwise="two.ways.backward", 
                      interactions.reg = TRUE,
                      min.variation = 0.5,
                      correlation = 0.9,
                      min.obs = 10)

save(GLMresultsD13vsD20, file = "Integration/GLMresultsD13vsD20.RData")
#load(file ="ntegration/GLMresultsD13vsD20.RData")

## Exploring results
head(GLMresultsD13vsD20$GlobalSummary$GoodnessOfFit) #Model p-vals. degrees of freedom og residuals, R-squared, AIC
head(GLMresultsD13vsD20$GlobalSummary$ReguPerGene) #For each omic/gene: number of initial regulators and significat regulators
head(GLMresultsD13vsD20$GlobalSummary$GenesNOmodel) #List of genes with no model obtained. #99 genes with no model

GLMresultsD13vsD20$arguments$edesign$Day = factor(GLMresultsD13vsD20$arguments$edesign$Day, levels = c("0", "7","13", "20"))

SigniRegulations = RegulationPerCondition(GLMresultsD13vsD20, betaTest = FALSE)
length(unique(SigniRegulations$gene)) # 1870 genes with significant regulators
table(SigniRegulations$omic) #5651 significant methylation regulators

# Plotting significant regulations ----------------------------------------

#Top 10 genes, TRP100
top10 = sigD13D20$Id[1:10]
mygene = top10[1]  
#par(mfrow = c(4, 3))
plotGLM(GLMresultsD13vsD20, gene = mygene, regulator = NULL, 
        xlab = "", reguValues = NULL, plotPerOmic = FALSE, order = FALSE,
        gene.col = 1, regu.col = "green4", replicates = TRUE, 
        cont.var = design2$Day)#, cond2plot = design2$Concentration.uM)

```

```{r function for calculation mean and sd}
data_summary <- function(data, varname, groupnames){
  require(plyr)
  summary_func <- function(x, col){
    c(mean = mean(x[[col]], na.rm=TRUE),
      sd = sd(x[[col]], na.rm=TRUE))
  }
  data_sum<-ddply(data, groupnames, .fun=summary_func,
                  varname)
  data_sum <- plyr::rename(data_sum,c("mean" = varname))
 return(data_sum)
}

betas2 = as.data.frame(betas)
betas2 = betas2[,keep]
betas2$cgid = rownames(betas2)
```

#Visualize MORE CpGs
```{r Lineplots MORE-sig CpGs}
#---PAX6--------------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000007372$significantRegulators

PAX6allCPGS = rownames(GLMresultsD0vsD20$ResultsPerGene$ENSG00000007372$allRegulators)

#PAX6, ENSG00000007372, sig: "cg01904582"
PAX6betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg01904582"))
PAX6betas = PAX6betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
PAX6_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000007372") 
PAX6_NC = PAX6_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

PAX6combined = merge(PAX6betas, PAX6_NC, by = "label")
PAX6combined = merge(PAX6combined, target, by = "label")


PAX6betas <- data_summary(PAX6combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))


a = ggplot(PAX6betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid ))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width=0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) + 
  theme_few()+
theme(legend.position= c(0.8,0.15),  legend.key.size = unit(0.1, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

PAX6counts <- data_summary(PAX6combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(PAX6counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width=0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")


grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("PAX6-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#VIM-------------------------------------------------------------------------------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000026025$significantRegulators

VIMallCPGS = rownames(GLMresultsD0vsD20$ResultsPerGene$ENSG00000026025$allRegulators)

#VIM, ENSG00000026025, sig: "cg04400217" "cg18876173"
VIMbetas = betas2 %>% dplyr::filter(., cgid  %in% c("cg04400217", "cg18876173"))
VIMbetas = VIMbetas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
VIM_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000026025") 
VIM_NC = VIM_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

VIMcombined = merge(VIMbetas, VIM_NC, by = "label")
VIMcombined = merge(VIMcombined, target, by = "label")


VIMbetas <- data_summary(VIMcombined, varname="Betas", 
                    groupnames=c("Day", "cgid"))

a = ggplot(VIMbetas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid,shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width=0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) + 
  theme_few()+
theme(legend.position= c(0.8,0.85), legend.key.size = unit(0.1, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

VIMcounts <- data_summary(VIMcombined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(VIMcounts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width=0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")


grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("VIM-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()


#MAP2----------------------------------------------------------------------------------------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000078018$significantRegulators

MAP2allCPGS = rownames(GLMresultsD0vsD20$ResultsPerGene$ENSG00000078018$allRegulators)

#MAP2, ENSG00000078018, sig: "cg15907018" "cg14361132" "cg15550769"
MAP2betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg15907018", "cg14361132", "cg15550769"))
MAP2betas = MAP2betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
MAP2_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000078018") 
MAP2_NC = MAP2_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

MAP2combined = merge(MAP2betas, MAP2_NC, by = "label")
MAP2combined = merge(MAP2combined, target, by = "label")

MAP2betas <- data_summary(MAP2combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))


a = ggplot(MAP2betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) + 
  theme_few()+
theme(legend.position= c(0.15,0.15), legend.key.size = unit(0.1, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

MAP2counts <- data_summary(MAP2combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(MAP2counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("MAP2-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#NANOG-------------------------------------------------------------------------------------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000111704$significantRegulators

NANOGallCPGS = rownames(GLMresultsD0vsD20$ResultsPerGene$ENSG00000111704$allRegulators)

#NANOG, ENSG00000111704, sig: "cg10660253" "cg18781032"
NANOGbetas = betas2 %>% dplyr::filter(., cgid  %in% c("cg10660253", "cg18781032"))
NANOGbetas = NANOGbetas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
NANOG_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000111704") 
NANOG_NC = NANOG_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

NANOGcombined = merge(NANOGbetas, NANOG_NC, by = "label")
NANOGcombined = merge(NANOGcombined, target, by = "label")


NANOGbetas <- data_summary(NANOGcombined, varname="Betas", 
                    groupnames=c("Day", "cgid"))


a = ggplot(NANOGbetas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid ))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18))+
  labs(x = "",y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) + 
  theme_few()+
theme(legend.position= c(0.8,0.15), legend.key.size = unit(0.1, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

NANOGcounts <- data_summary(NANOGcombined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(NANOGcounts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("NANOG-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()


#NES-------------------------------------------------------------------------------------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000132688$significantRegulators

NESallCPGS = rownames(GLMresultsD0vsD20$ResultsPerGene$ENSG00000132688$allRegulators)

#NES, ENSG00000132688, sig: "cg17872853" "cg04234543" "cg04671462" "cg12080906" "cg26706311"
NESbetas = betas2 %>% dplyr::filter(., cgid  %in% c("cg17872853", "cg04234543", "cg04671462" ,"cg12080906" ,"cg26706311"))
NESbetas = NESbetas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
NES_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000132688") 
NES_NC = NES_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

NEScombined = merge(NESbetas, NES_NC, by = "label")
NEScombined = merge(NEScombined, target, by = "label")


NESbetas <- data_summary(NEScombined, varname="Betas", 
                    groupnames=c("Day", "cgid"))


a = ggplot(NESbetas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "",y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) + 
  theme_few()+
theme(legend.position= c(0.15,0.85), legend.key.size = unit(0.1, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))



NEScounts <- data_summary(NEScombined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(NEScounts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("NES-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#NEUROD1-------------------------------------------------------------------------------------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000162992$significantRegulators

NEUROD1allCPGS = rownames(GLMresultsD0vsD20$ResultsPerGene$ENSG00000162992$allRegulators)

#NEUROD1, ENSG00000132688, sig: "cg03767531" "cg16640855"
NEUROD1betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg03767531", "cg16640855"))
NEUROD1betas = NEUROD1betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
NEUROD1_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000162992") 
NEUROD1_NC = NEUROD1_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

NEUROD1combined = merge(NEUROD1betas, NEUROD1_NC, by = "label")
NEUROD1combined = merge(NEUROD1combined, target, by = "label")


NEUROD1betas <- data_summary(NEUROD1combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))


a = ggplot(NEUROD1betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid ))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) + 
  theme_few()+
theme(legend.position= c(0.2,0.85), legend.key.size = unit(0.1, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))



NEUROD1counts <- data_summary(NEUROD1combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(NEUROD1counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")


grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("NEUROD1-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#OTX2-------------------------------------------------------------------------------------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000165588$significantRegulators

OTX2allCPGS = rownames(GLMresultsD0vsD20$ResultsPerGene$ENSG00000165588$allRegulators)

#OTX2, ENSG00000165588, sig: "cg24748769" "cg19895517"
OTX2betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg24748769" ,"cg19895517"))
OTX2betas = OTX2betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
OTX2_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000165588") 
OTX2_NC = OTX2_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

OTX2combined = merge(OTX2betas, OTX2_NC, by = "label")
OTX2combined = merge(OTX2combined, target, by = "label")


OTX2betas <- data_summary(OTX2combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))


a = ggplot(OTX2betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid ))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) + 
  theme_few()+
theme(legend.position= c(0.8,0.1), legend.key.size = unit(0.1, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))


OTX2counts <- data_summary(OTX2combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(OTX2counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("OTX2-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()


#FOXG1-------------------------------------------------------------------------------------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000176165$significantRegulators

FOXG1allCPGS = rownames(GLMresultsD0vsD20$ResultsPerGene$ENSG00000176165$allRegulators)

#FOXG1, ENSG00000176165, sig: "cg05872306" "cg19786627"
FOXG1betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg05872306", "cg19786627"))
FOXG1betas = FOXG1betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
FOXG1_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000176165") 
FOXG1_NC = FOXG1_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

FOXG1combined = merge(FOXG1betas, FOXG1_NC, by = "label")
FOXG1combined = merge(FOXG1combined, target, by = "label")

FOXG1betas <- data_summary(FOXG1combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))

a = ggplot(FOXG1betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) + 
  theme_few()+
theme(legend.position= c(0.8,0.85), legend.key.size = unit(0.1, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

FOXG1counts <- data_summary(FOXG1combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(FOXG1counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("FOXG1-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#SOX2-------------------------------------------------------------------------------------------------------
GLMresultsDay7vsDay0$ResultsPerGene$ENSG00000181449$significantRegulators

SOX2allCPGS = rownames(GLMresultsDay7vsDay0$ResultsPerGene$ENSG00000181449$allRegulators)

#SOX2, ENSG00000181449, sig: "cg08062338"
SOX2betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg08062338"))
SOX2betas = SOX2betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
SOX2_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000181449") 
SOX2_NC = SOX2_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

SOX2combined = merge(SOX2betas, SOX2_NC, by = "label")
SOX2combined = merge(SOX2combined, target, by = "label")

SOX2betas <- data_summary(SOX2combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))


a = ggplot(SOX2betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) + 
  theme_few()+
theme(legend.position= c(0.8,0.85), legend.key.size = unit(0.1, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

SOX2counts <- data_summary(SOX2combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(SOX2counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  scale_y_continuous(limits=c(0, 57000)) +
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("SOX2-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#SOX1-------------------------------------------------------------------------------------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000182968$significantRegulators

SOX1allCPGS = rownames(GLMresultsD0vsD20$ResultsPerGene$ENSG00000182968$allRegulators)

#SOX1, ENSG00000181449, sig: "cg10245273" "cg01757312" "cg19407095"
SOX1betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg10245273", "cg01757312", "cg19407095"))
SOX1betas = SOX1betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
SOX1_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000182968") 
SOX1_NC = SOX1_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

SOX1combined = merge(SOX1betas, SOX1_NC, by = "label")
SOX1combined = merge(SOX1combined, target, by = "label")

SOX1betas <- data_summary(SOX1combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))

a = ggplot(SOX1betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,
                 position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) + 
  theme_few()+
theme(legend.position= c(0.8,0.85), legend.key.size = unit(0.1, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

SOX1counts <- data_summary(SOX1combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(SOX1counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("SOX1-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#POU5F1-------------------------------------------------------------------------------------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000204531$significantRegulators

POU5F1allCPGS = rownames(GLMresultsD0vsD20$ResultsPerGene$ENSG00000204531$allRegulators)

#POU5F1, ENSG00000181449, sig: "cg13564588" "cg00357273" "cg04099091" "cg09031041" "cg13083810" "cg15278102" "cg15948871" "cg16626479" "cg17364250"
POU5F1betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg13564588", "cg00357273", "cg04099091", "cg09031041", "cg13083810", "cg15278102", "cg15948871", "cg16626479", "cg17364250"))
POU5F1betas = POU5F1betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
POU5F1_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000204531") 
POU5F1_NC = POU5F1_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

POU5F1combined = merge(POU5F1betas, POU5F1_NC, by = "label")
POU5F1combined = merge(POU5F1combined, target, by = "label")

POU5F1betas <- data_summary(POU5F1combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))


a = ggplot(POU5F1betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape=cgid))+
  geom_line() + 
  geom_point(size = 2)+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,
                 position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) + 
  theme_few()+
theme(legend.position= c(0.12,0.82), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

POU5F1counts <- data_summary(POU5F1combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(POU5F1counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("POU5F1-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#TUBB3-------------------------------------------------------------------------------------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000258947$significantRegulators
TUBB3allCPGS = rownames(GLMresultsD0vsD20$ResultsPerGene$ENSG00000258947$allRegulators)

#TUBB3, ENSG00000181449, sig: "cg09826306" "cg13006202" "cg14078385" "cg15669092" "cg18701351" "cg20776939" "cg26892562" "cg27170458"
TUBB3betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg09826306", "cg13006202", "cg14078385", "cg15669092", "cg18701351", "cg20776939", "cg26892562", "cg27170458"))
TUBB3betas = TUBB3betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
TUBB3_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000258947") 
TUBB3_NC = TUBB3_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

TUBB3combined = merge(TUBB3betas, TUBB3_NC, by = "label")
TUBB3combined = merge(TUBB3combined, target, by = "label")

TUBB3betas <- data_summary(TUBB3combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))

a = ggplot(TUBB3betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,
                 position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) + 
  theme_few()+
theme(legend.position= c(0.15,0.85), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

TUBB3counts <- data_summary(TUBB3combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(TUBB3counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("TUBB3-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()
```



```{r Manual plots GeneList}

#NEFH------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000100285$significantRegulators
NEFHallCPGS = rownames(GLMresultsGeneList$ResultsPerGene$ENSG00000100285$allRegulators)

#NEFH,ENSG00000100285 "cg18129621" "cg02994956"
NEFHbetas = betas2 %>% dplyr::filter(., cgid  %in% c("cg18129621", "cg02994956"))
NEFHbetas = NEFHbetas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
NEFH_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000100285") 
NEFH_NC = NEFH_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

NEFHcombined = merge(NEFHbetas, NEFH_NC, by = "label")
NEFHcombined = merge(NEFHcombined, target, by = "label")

NEFHbetas <- data_summary(NEFHcombined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(NEFHbetas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.85), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

NEFHcounts <- data_summary(NEFHcombined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(NEFHcounts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("NEFH-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#KIF5A-----------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000155980$significantRegulators
KIF5AallCPGS = rownames(GLMresultsGeneList$ResultsPerGene$ENSG00000155980$allRegulators)

#KIF5A,ENSG00000139970 "cg05024573" "cg06667339" "cg16949674" "cg17083982" "cg18628483" "cg23186534" "cg26386962"
KIF5Abetas = betas2 %>% dplyr::filter(., cgid  %in% c("cg05024573", "cg06667339" ,"cg16949674", "cg17083982", "cg18628483", "cg23186534" ,"cg26386962"))
KIF5Abetas = KIF5Abetas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
KIF5A_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000155980") 
KIF5A_NC = KIF5A_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

KIF5Acombined = merge(KIF5Abetas, KIF5A_NC, by = "label")
KIF5Acombined = merge(KIF5Acombined, target, by = "label")

KIF5Abetas <- data_summary(KIF5Acombined, varname="Betas", 
                    groupnames=c("Day", "cgid"))

a = ggplot(KIF5Abetas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.15), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))


KIF5Acounts <- data_summary(KIF5Acombined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(KIF5Acounts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0, position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("KIF5A-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()


#GNG11-----------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000127920$significantRegulators
GNG11allCPGS = rownames(GLMresultsGeneList$ResultsPerGene$ENSG00000127920$allRegulators)

#GNG11,ENSG00000127920 "cg00536924" "cg08038054" "cg08236022" "cg25546811"
GNG11betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg00536924", "cg08038054", "cg08236022", "cg25546811"))
GNG11betas = GNG11betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
GNG11_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000127920") 
GNG11_NC = GNG11_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

GNG11combined = merge(GNG11betas, GNG11_NC, by = "label")
GNG11combined = merge(GNG11combined, target, by = "label")

GNG11betas <- data_summary(GNG11combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))

a = ggplot(GNG11betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.5), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

GNG11counts <- data_summary(GNG11combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(GNG11counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("GNG11-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#ERBB3-----------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000065361$significantRegulators
ERBB3allCPGS = rownames(GLMresultsGeneList$ResultsPerGene$ENSG00000065361$allRegulators)

#ERBB3,ENSG00000065361 "cg27488983"     "cg00907267"     "cg26344379"     "ch.12.1171804F"
ERBB3betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg27488983","cg00907267","cg26344379"))
ERBB3betas = ERBB3betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
ERBB3_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000065361") 
ERBB3_NC = ERBB3_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

ERBB3combined = merge(ERBB3betas, ERBB3_NC, by = "label")
ERBB3combined = merge(ERBB3combined, target, by = "label")

ERBB3betas <- data_summary(ERBB3combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))

a = ggplot(ERBB3betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.85), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

ERBB3counts <- data_summary(ERBB3combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(ERBB3counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("ERBB3-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#TRDN-----------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000186439$significantRegulators
TRDNallCPGS = rownames(GLMresultsGeneList$ResultsPerGene$ENSG00000186439$allRegulators)

#TRDN, ENSG00000186439, "cg05897020" "cg08690174" "cg17889149" "cg19553563" "cg24153143"
TRDNbetas = betas2 %>% dplyr::filter(., cgid  %in% c("cg05897020", "cg08690174", "cg17889149", "cg19553563", "cg24153143"))
TRDNbetas = TRDNbetas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
TRDN_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000186439") 
TRDN_NC = TRDN_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

TRDNcombined = merge(TRDNbetas, TRDN_NC, by = "label")
TRDNcombined = merge(TRDNcombined, target, by = "label")


TRDNbetas <- data_summary(TRDNcombined, varname="Betas", 
                    groupnames=c("Day", "cgid"))

a = ggplot(TRDNbetas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.15), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

TRDNcounts <- data_summary(TRDNcombined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(TRDNcounts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("TRDN-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#PRSS23-----------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000150687$significantRegulators
PRSS23allCPGS = rownames(GLMresultsGeneList$ResultsPerGene$ENSG00000150687$allRegulators)

#PRSS23, ENSG00000150687,"cg09376465" "cg10453532" "cg20216069" "cg05385995" "cg20400693"
PRSS23betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg09376465", "cg10453532", "cg20216069", "cg05385995", "cg20400693"))
PRSS23betas = PRSS23betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
PRSS23_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000150687") 
PRSS23_NC = PRSS23_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

PRSS23combined = merge(PRSS23betas, PRSS23_NC, by = "label")
PRSS23combined = merge(PRSS23combined, target, by = "label")

PRSS23betas <- data_summary(PRSS23combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))

a = ggplot(PRSS23betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.8,0.15), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

PRSS23counts <- data_summary(PRSS23combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(PRSS23counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("PRSS23-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-FLT1----------------------------

GLMresultsD0vsD20$ResultsPerGene$ENSG00000102755$significantRegulators

#FLT1, ENSG00000102755,"cg05961406" "cg16977411" "cg24850711"
FLT1betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg05961406", "cg16977411", "cg24850711"))
FLT1betas = FLT1betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
FLT1_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000102755") 
FLT1_NC = FLT1_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

FLT1combined = merge(FLT1betas, FLT1_NC, by = "label")
FLT1combined = merge(FLT1combined, target, by = "label")

FLT1betas <- data_summary(FLT1combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))

a = ggplot(FLT1betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.8,0.15), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

FLT1counts <- data_summary(FLT1combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(FLT1counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("FLT1-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-MAP7----------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000135525$significantRegulators

#MAP7, ENSG00000135525,"ch.6.2623783F" "cg13313598"    "cg21471225"    "cg25667077"  
MAP7betas = betas2 %>% dplyr::filter(., cgid  %in% c("ch.6.2623783F", "cg13313598","cg21471225","cg25667077"))
MAP7betas = MAP7betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
MAP7_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000135525") 
MAP7_NC = MAP7_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

MAP7combined = merge(MAP7betas, MAP7_NC, by = "label")
MAP7combined = merge(MAP7combined, target, by = "label")

MAP7betas <- data_summary(MAP7combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))

a = ggplot(MAP7betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.15), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

MAP7counts <- data_summary(MAP7combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(MAP7counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("MAP7-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#NRG1-----------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000157168$significantRegulators

#NRG1, ENSG00000157168, sig:"cg11454563" "cg07321819" "cg02640525" "cg12954529" "cg13618529" "cg14598943"
NRG1betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg11454563", "cg07321819", "cg02640525", "cg12954529", "cg13618529", "cg14598943"))
NRG1betas = NRG1betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
NRG1_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000157168") 
NRG1_NC = NRG1_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

NRG1combined = merge(NRG1betas, NRG1_NC, by = "label")
NRG1combined = merge(NRG1combined, target, by = "label")

NRG1betas <- data_summary(NRG1combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(NRG1betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.5), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

NRG1counts <- data_summary(NRG1combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(NRG1counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("NRG1-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#REEP1-----------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000068615$significantRegulators

#REEP1, ENSG00000068615, sig:"cg16934445" "cg06024918" "cg18946805" "cg27010375" "cg27359670"
REEP1betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg16934445", "cg06024918", "cg18946805", "cg27010375", "cg27359670"))
REEP1betas = REEP1betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
REEP1_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000068615") 
REEP1_NC = REEP1_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

REEP1combined = merge(REEP1betas, REEP1_NC, by = "label")
REEP1combined = merge(REEP1combined, target, by = "label")

REEP1betas <- data_summary(REEP1combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(REEP1betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.2), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

REEP1counts <- data_summary(REEP1combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(REEP1counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("REEP1-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-DCLK2----------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000170390$significantRegulators

#DCLK2, ENSG00000170390, sig:cg25674620" "cg14948279" "cg20248790" "cg10272520" "cg12281380" "cg18426033" "cg18969798" "cg27188589"
DCLK2betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg25674620", "cg14948279", "cg20248790", "cg10272520", "cg12281380", "cg18426033", "cg18969798", "cg27188589"))
DCLK2betas = DCLK2betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
DCLK2_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000170390") 
DCLK2_NC = DCLK2_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

DCLK2combined = merge(DCLK2betas, DCLK2_NC, by = "label")
DCLK2combined = merge(DCLK2combined, target, by = "label")

DCLK2betas <- data_summary(DCLK2combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))

a = ggplot(DCLK2betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.3), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

DCLK2counts <- data_summary(DCLK2combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(DCLK2counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("DCLK2-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-EYA4----------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000112319$significantRegulators

#EYA4, ENSG00000112319, sig:"cg25075347" "cg10782703" "cg06966404" "cg07224291" "cg12162340" "cg13460476" "cg23028941"
EYA4betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg25075347", "cg10782703", "cg06966404", "cg07224291", "cg12162340", "cg13460476", "cg23028941"))
EYA4betas = EYA4betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
EYA4_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000112319") 
EYA4_NC = EYA4_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

EYA4combined = merge(EYA4betas, EYA4_NC, by = "label")
EYA4combined = merge(EYA4combined, target, by = "label")

EYA4betas <- data_summary(EYA4combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(EYA4betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.3), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

EYA4counts <- data_summary(EYA4combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(EYA4counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("EYA4-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-ZEB2----------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000169554$significantRegulators

#ZEB2, ENSG00000169554, sig:"cg08754046" "cg00025331" "cg03303633" "cg14807945" "cg16219453" "cg22267613"
ZEB2betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg08754046", "cg00025331", "cg03303633", "cg14807945", "cg16219453" ,"cg22267613"))
ZEB2betas = ZEB2betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
ZEB2_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000169554") 
ZEB2_NC = ZEB2_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

ZEB2combined = merge(ZEB2betas, ZEB2_NC, by = "label")
ZEB2combined = merge(ZEB2combined, target, by = "label")

ZEB2betas <- data_summary(ZEB2combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(ZEB2betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.5), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

ZEB2counts <- data_summary(ZEB2combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(ZEB2counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("ZEB2-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-NPAS3----------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000151322$significantRegulators

#NPAS3, ENSG00000151322, sig:"cg16250278" "cg00972222" "cg07050194" "cg10785997" "cg11793860" "cg13496915" "cg16661114" "cg18948852" "cg20925873" "cg26195358"
NPAS3betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg16250278", "cg00972222", "cg07050194", "cg10785997", "cg11793860" ,"cg13496915", "cg16661114", "cg18948852", "cg20925873", "cg26195358"))
NPAS3betas = NPAS3betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
NPAS3_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000151322") 
NPAS3_NC = NPAS3_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

NPAS3combined = merge(NPAS3betas, NPAS3_NC, by = "label")
NPAS3combined = merge(NPAS3combined, target, by = "label")

NPAS3betas <- data_summary(NPAS3combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(NPAS3betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1", "#666666"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= "top", legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

NPAS3counts <- data_summary(NPAS3combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(NPAS3counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("NPAS3-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#DHCR24-----------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000116133$significantRegulators

#DHCR24, ENSG00000116133, sig:"cg17841473" "cg17901584" "cg20291423"
DHCR24betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg17841473", "cg17901584", "cg20291423"))
DHCR24betas = DHCR24betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
DHCR24_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000116133") 
DHCR24_NC = DHCR24_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

DHCR24combined = merge(DHCR24betas, DHCR24_NC, by = "label")
DHCR24combined = merge(DHCR24combined, target, by = "label")

DHCR24betas <- data_summary(DHCR24combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(DHCR24betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,
                 position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.2,0.2), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

DHCR24counts <- data_summary(DHCR24combined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(DHCR24counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("DHCR24-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-CDH11----------------------------
GLMresultsDay7vsDay0$ResultsPerGene$ENSG00000140937$significantRegulators

#CDH11, ENSG00000140937, sig:"cg07951949" "cg09108073" "cg01021552" "cg10015051" "cg21458849"
CDH11betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg07951949", "cg09108073", "cg01021552", "cg10015051", "cg21458849"))
CDH11betas = CDH11betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
CDH11_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000140937") 
CDH11_NC = CDH11_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

CDH11combined = merge(CDH11betas, CDH11_NC, by = "label")
CDH11combined = merge(CDH11combined, target, by = "label")

CDH11betas <- data_summary(CDH11combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(CDH11betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.2,0.2), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

CDH11counts <- data_summary(CDH11combined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(CDH11counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("CDH11-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#DNMT3B-----------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000088305$significantRegulators

#DNMT3B, ENSG00000088305, sig: [1] "cg01281600" "cg09135144" "cg09835408" "cg14224313" "cg17475857" "cg17482740" "cg17697897", "cg19184414" "cg22052056" "cg22605822" "cg24403338" "cg26553763"
DNMT3Bbetas = betas2 %>% dplyr::filter(., cgid  %in% c("cg01281600", "cg09135144", "cg09835408", "cg14224313", "cg17475857", "cg17482740" ,"cg17697897", "cg19184414", "cg22052056", "cg22605822", "cg24403338", "cg26553763"))
DNMT3Bbetas = DNMT3Bbetas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
DNMT3B_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000088305") 
DNMT3B_NC = DNMT3B_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

DNMT3Bcombined = merge(DNMT3Bbetas, DNMT3B_NC, by = "label")
DNMT3Bcombined = merge(DNMT3Bcombined, target, by = "label")

DNMT3Bbetas <- data_summary(DNMT3Bcombined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(DNMT3Bbetas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +
  theme_few()+
theme(legend.position= "top", legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 5), legend.title = element_text(size = 7))

DNMT3Bcounts <- data_summary(DNMT3Bcombined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(DNMT3Bcounts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("DNMT3B-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-CDH1----------------------------------------------------------------------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000039068$significantRegulators

#CDH1, ENSG00000039068, sig: "cg07762788" "cg05303053" "cg00632260" "cg08051386" "cg13947099" "cg15653892" "cg19834745" "cg20750889" "cg23166227" "cg25218831" "cg26508465"
CDH1betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg07762788", "cg05303053", "cg00632260", "cg08051386", "cg13947099", "cg15653892", "cg19834745", "cg20750889", "cg23166227", "cg25218831", "cg26508465"))
CDH1betas = CDH1betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
CDH1_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000039068") 
CDH1_NC = CDH1_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

CDH1combined = merge(CDH1betas, CDH1_NC, by = "label")
CDH1combined = merge(CDH1combined, target, by = "label")

CDH1betas <- data_summary(CDH1combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(CDH1betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.8,0.4), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

CDH1counts <- data_summary(CDH1combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(CDH1counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("CDH1-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#---------------------------------------------------------------------REST----------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000084093$significantRegulators

#REST, ENSG00000084093, sig: "cg07964006" "cg14540093"
RESTbetas = betas2 %>% dplyr::filter(., cgid  %in% c("cg07964006", "cg14540093"))
RESTbetas = RESTbetas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
REST_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000084093") 
REST_NC = REST_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

RESTcombined = merge(RESTbetas, REST_NC, by = "label")
RESTcombined = merge(RESTcombined, target, by = "label")

RESTbetas <- data_summary(RESTcombined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(RESTbetas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC"))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.2,0.8), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

RESTcounts <- data_summary(RESTcombined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(RESTcounts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  scale_y_continuous(limits=c(0, 12000)) + 
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("REST-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-------------------------------------------------------------------------CHD2-----------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000170558$significantRegulators

#CDH2, ENSG00000170558, sig: "cg18666853" "cg15322562" "cg23244545" "cg15859602"
CDH2betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg18666853", "cg15322562", "cg23244545", "cg15859602"))
CDH2betas = CDH2betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
CDH2_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000170558") 
CDH2_NC = CDH2_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

CDH2combined = merge(CDH2betas, CDH2_NC, by = "label")
CDH2combined = merge(CDH2combined, target, by = "label")

CDH2betas <- data_summary(CDH2combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(CDH2betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.2,0.15), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

CDH2counts <- data_summary(CDH2combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(CDH2counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  scale_y_continuous(limits=c(0, NA)) +
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("CDH2-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-----------------------------------------------------------------------------GAL---------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000069482$significantRegulators

#GAL, ENSG00000069482, sig: "cg22007237" "cg01545145" "cg23009355"
GALbetas = betas2 %>% dplyr::filter(., cgid  %in% c("cg22007237", "cg01545145", "cg23009355"))
GALbetas = GALbetas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
GAL_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000069482") 
GAL_NC = GAL_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

GALcombined = merge(GALbetas, GAL_NC, by = "label")
GALcombined = merge(GALcombined, target, by = "label")

GALbetas <- data_summary(GALcombined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(GALbetas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
  scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.2), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

GALcounts <- data_summary(GALcombined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(GALcounts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("GAL-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#--------------------------------------------------------------------------------------------EPCAM--------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000119888$significantRegulators

#EPCAM, ENSG00000119888, sig:"cg07658533" "cg03210866" "cg03706175" "cg16076328"
EPCAMbetas = betas2 %>% dplyr::filter(., cgid  %in% c("cg07658533", "cg03210866", "cg03706175", "cg16076328"))
EPCAMbetas = EPCAMbetas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
EPCAM_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000119888") 
EPCAM_NC = EPCAM_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

EPCAMcombined = merge(EPCAMbetas, EPCAM_NC, by = "label")
EPCAMcombined = merge(EPCAMcombined, target, by = "label")

EPCAMbetas <- data_summary(EPCAMcombined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(EPCAMbetas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0, position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.85), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 6), legend.title = element_text(size = 8))

EPCAMcounts <- data_summary(EPCAMcombined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(EPCAMcounts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("EPCAM-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#----------------------------------------------------------------ID1------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000125968$significantRegulators

#ID1, ENSG00000125968, sig:"cg05194250"
ID1betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg05194250"))
ID1betas = ID1betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
ID1_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000125968") 
ID1_NC = ID1_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

ID1combined = merge(ID1betas, ID1_NC, by = "label")
ID1combined = merge(ID1combined, target, by = "label")

ID1betas <- data_summary(ID1combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(ID1betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,
                 position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.85), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 5), legend.title = element_text(size = 7))

ID1counts <- data_summary(ID1combined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(ID1counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("ID1-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()


#-------------ID3----------------------------------------------------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000117318$significantRegulators

#ID3, ENSG00000117318, sig:"cg02343604"
ID3betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg02343604"))
ID3betas = ID3betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
ID3_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000117318") 
ID3_NC = ID3_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

ID3combined = merge(ID3betas, ID3_NC, by = "label")
ID3combined = merge(ID3combined, target, by = "label")


ID3betas <- data_summary(ID3combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))


a = ggplot(ID3betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.85), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 5), legend.title = element_text(size = 7))

ID3counts <- data_summary(ID3combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(ID3counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("ID3-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()


#-----------------------------------------------------------------ID4-----
GLMresultsD0vsD20$ResultsPerGene$ENSG00000172201$significantRegulators

#ID4, ENSG00000172201, sig:"cg04359753"
ID4betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg04359753"))
ID4betas = ID4betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
ID4_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000172201") 
ID4_NC = ID4_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

ID4combined = merge(ID4betas, ID4_NC, by = "label")
ID4combined = merge(ID4combined, target, by = "label")


ID4betas <- data_summary(ID4combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))

a = ggplot(ID4betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,
                 position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC"))+
   scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.85), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 5), legend.title = element_text(size = 7))

ID4counts <- data_summary(ID4combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(ID4counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("ID4-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-------------------------------------------------------------NKX2_1---------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000136352$significantRegulators

#NKX2_1, ENSG00000136352, sig:"cg21282193" "cg01652665" "cg14290904"
NKX2_1betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg21282193", "cg01652665", "cg14290904"))
NKX2_1betas = NKX2_1betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
NKX2_1_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000136352") 
NKX2_1_NC = NKX2_1_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

NKX2_1combined = merge(NKX2_1betas, NKX2_1_NC, by = "label")
NKX2_1combined = merge(NKX2_1combined, target, by = "label")

NKX2_1betas <- data_summary(NKX2_1combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))


a = ggplot(NKX2_1betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.85,0.9), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 5), legend.title = element_text(size = 7))

NKX2_1counts <- data_summary(NKX2_1combined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(NKX2_1counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("NKX2.1-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#----------------------------------------------------------------NR6A1------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000148200$significantRegulators

#NR6A1, ENSG00000148200, sig:"cg06321925" "cg08224390" "cg12954102" "cg13539372" "cg14156446" "cg15096615" "cg16717651"
NR6A1betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg06321925", "cg08224390", "cg12954102", "cg13539372", "cg14156446", "cg15096615", "cg16717651"))
NR6A1betas = NR6A1betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
NR6A1_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000148200") 
NR6A1_NC = NR6A1_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

NR6A1combined = merge(NR6A1betas, NR6A1_NC, by = "label")
NR6A1combined = merge(NR6A1combined, target, by = "label")

NR6A1betas <- data_summary(NR6A1combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))

a = ggplot(NR6A1betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.1,0.15), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 5), legend.title = element_text(size = 7))

NR6A1counts <- data_summary(NR6A1combined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(NR6A1counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("NR6A1-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#----------------------------------------------------------------SOX6------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000110693$significantRegulators

#SOX6, ENSG00000110693, sig:"cg08643003" "cg26830200" "cg00664860" "cg08224625" "cg09749669" "cg10825530" "cg15497474" "cg21992400" "cg23530266" "cg24331727"
SOX6betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg08643003", "cg26830200", "cg00664860", "cg08224625", "cg09749669", "cg10825530", "cg15497474", "cg21992400", "cg23530266", "cg24331727"))
SOX6betas = SOX6betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
SOX6_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000110693") 
SOX6_NC = SOX6_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

SOX6combined = merge(SOX6betas, SOX6_NC, by = "label")
SOX6combined = merge(SOX6combined, target, by = "label")

SOX6betas <- data_summary(SOX6combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(SOX6betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.1,0.3), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.1,"cm"), legend.text = element_text(size = 5), legend.title = element_text(size = 7))

SOX6counts <- data_summary(SOX6combined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(SOX6counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("SOX6-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#----------------------------------------------------------------LHX2------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000106689$significantRegulators

#LHX2, ENSG00000106689, sig:"cg19264651" "cg06692990" "cg00982919" "cg03306024" "cg03559209" "cg03709033" "cg04094300" "cg05777357" "cg05900009" "cg06890720" "cg12002589" "cg13342881" "cg13585776" "cg14071579" "cg14425564" "cg16427654" "cg20939662" "cg21195468"
LHX2betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg19264651", "cg06692990", "cg00982919", "cg03306024", "cg03559209", "cg03709033", "cg04094300", "cg05777357", "cg05900009", "cg06890720", "cg12002589", "cg13342881", "cg13585776", "cg14071579", "cg14425564", "cg16427654", "cg20939662", "cg21195468"))
LHX2betas = LHX2betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
LHX2_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000106689") 
LHX2_NC = LHX2_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

LHX2combined = merge(LHX2betas, LHX2_NC, by = "label")
LHX2combined = merge(LHX2combined, target, by = "label")

LHX2betas <- data_summary(LHX2combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(LHX2betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= "top", legend.key.size = unit(0.005, "cm"), legend.key.width = unit(0.005,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

LHX2counts <- data_summary(LHX2combined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(LHX2counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("LHX2-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#--------------------------------------------------------------ZIC4--------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000174963$significantRegulators

#ZIC4, ENSG00000174963, sig:"cg04079301" "cg20939084" "cg00896370" "cg02820514" "cg08393041" "cg13897134" "cg15105326" "cg16768018" "cg16790847" "cg17569743" "cg25449440" "cg26386385" "cg26790247"
ZIC4betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg04079301", "cg20939084", "cg00896370", "cg02820514", "cg08393041", "cg13897134", "cg15105326", "cg16768018", "cg16790847", "cg17569743", "cg25449440", "cg26386385", "cg26790247"))
ZIC4betas = ZIC4betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
ZIC4_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000174963") 
ZIC4_NC = ZIC4_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

ZIC4combined = merge(ZIC4betas, ZIC4_NC, by = "label")
ZIC4combined = merge(ZIC4combined, target, by = "label")

ZIC4betas <- data_summary(ZIC4combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(ZIC4betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.85,0.85), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.01,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

ZIC4counts <- data_summary(ZIC4combined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(ZIC4counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("ZIC4-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#---------------------------------------------------------------FOXH1-------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000160973$significantRegulators

#FOXH1, ENSG00000160973, sig:"cg03667317" "cg04515567" "cg07044391" "cg07205791" "cg13159060" "cg21293425"
FOXH1betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg03667317", "cg04515567", "cg07044391", "cg07205791", "cg13159060", "cg21293425"))
FOXH1betas = FOXH1betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
FOXH1_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000160973") 
FOXH1_NC = FOXH1_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

FOXH1combined = merge(FOXH1betas, FOXH1_NC, by = "label")
FOXH1combined = merge(FOXH1combined, target, by = "label")

FOXH1betas <- data_summary(FOXH1combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(FOXH1betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,
                 position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.1,0.90), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.01,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

FOXH1counts <- data_summary(FOXH1combined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(FOXH1counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("FOXH1-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-----------------------------------------------------------------DPPA4-----
GLMresultsD0vsD20$ResultsPerGene$ENSG00000121570$significantRegulators

#DPPA4, ENSG00000121570, sig:"cg19250607" "cg00487142" "cg07253829" "cg12443195" "cg13358761" "cg14836960" "cg18560366"
DPPA4betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg19250607", "cg00487142", "cg07253829", "cg12443195", "cg13358761", "cg14836960", "cg18560366"))
DPPA4betas = DPPA4betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
DPPA4_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000121570") 
DPPA4_NC = DPPA4_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

DPPA4combined = merge(DPPA4betas, DPPA4_NC, by = "label")
DPPA4combined = merge(DPPA4combined, target, by = "label")

DPPA4betas <- data_summary(DPPA4combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(DPPA4betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.85), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.01,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

DPPA4counts <- data_summary(DPPA4combined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(DPPA4counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,position=position_dodge(0.05) )+
  scale_y_continuous(limits=c(0, 60000)) + 
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("DPPA4-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#------------------------------------------------------------------FGF8----
GLMresultsD0vsD20$ResultsPerGene$ENSG00000107831$significantRegulators

#FGF8, ENSG00000107831, sig:"cg01374979" "cg04645545" "cg07037112" "cg16079364" "cg17633028" "cg22239593" "cg26820037"
FGF8betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg01374979", "cg04645545", "cg07037112", "cg16079364", "cg17633028", "cg22239593" ,"cg26820037"))
FGF8betas = FGF8betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
FGF8_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000107831") 
FGF8_NC = FGF8_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

FGF8combined = merge(FGF8betas, FGF8_NC, by = "label")
FGF8combined = merge(FGF8combined, target, by = "label")

FGF8betas <- data_summary(FGF8combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(FGF8betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,
                 position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.85), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.01,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

FGF8counts <- data_summary(FGF8combined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(FGF8counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("FGF8-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#------------------------------------------------------------------L1CAM----
GLMresultsD0vsD20$ResultsPerGene$ENSG00000198910$significantRegulators

L1CAMallCPGS = rownames(GLMresultsD0vsD20$ResultsPerGene$ENSG00000198910$allRegulators)

#L1CAM, ENSG00000198910, sig:"cg15794769"
L1CAMbetas = betas2 %>% dplyr::filter(., cgid  %in% c("cg15794769"))
L1CAMbetas = L1CAMbetas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
L1CAM_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000198910") 
L1CAM_NC = L1CAM_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

L1CAMcombined = merge(L1CAMbetas, L1CAM_NC, by = "label")
L1CAMcombined = merge(L1CAMcombined, target, by = "label")

L1CAMbetas <- data_summary(L1CAMcombined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(L1CAMbetas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.1,0.3), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.01,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

L1CAMcounts <- data_summary(L1CAMcombined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(L1CAMcounts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("L1CAM-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#------------------------------------------------------------------ELAVL4----
GLMresultsD0vsD20$ResultsPerGene$ENSG00000162374$significantRegulators

#ELAVL4, ENSG00000162374, sig:"cg03462926" "cg12227258"
ELAVL4betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg03462926", "cg12227258"))
ELAVL4betas = ELAVL4betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
ELAVL4_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000162374") 
ELAVL4_NC = ELAVL4_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

ELAVL4combined = merge(ELAVL4betas, ELAVL4_NC, by = "label")
ELAVL4combined = merge(ELAVL4combined, target, by = "label")

ELAVL4betas <- data_summary(ELAVL4combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(ELAVL4betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.15), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.01,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

ELAVL4counts <- data_summary(ELAVL4combined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(ELAVL4counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("ELAVL4-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()


#------------------------------------------------------------------DCX----
GLMresultsD0vsD20$ResultsPerGene$ENSG00000077279$significantRegulators

#DCX, ENSG00000077279, sig:"cg15334207" "cg27175910" "cg11031647" "cg13936874"
DCXbetas = betas2 %>% dplyr::filter(., cgid  %in% c("cg15334207", "cg27175910", "cg11031647", "cg13936874"))
DCXbetas = DCXbetas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
DCX_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000077279") 
DCX_NC = DCX_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

DCXcombined = merge(DCXbetas, DCX_NC, by = "label")
DCXcombined = merge(DCXcombined, target, by = "label")

DCXbetas <- data_summary(DCXcombined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(DCXbetas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,
                 position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.1,0.2), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.01,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

DCXcounts <- data_summary(DCXcombined, varname="NormalizedCounts", 
                    groupnames=c("Day", "Id"))

b = ggplot(DCXcounts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("DCX-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#------------------------------------------------------------------SCG2----
GLMresultsD0vsD20$ResultsPerGene$ENSG00000171951$significantRegulators

#SCG2, ENSG00000171951, sig:"cg18840330" "cg21616047"
SCG2betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg18840330", "cg21616047"))
SCG2betas = SCG2betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
SCG2_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000171951") 
SCG2_NC = SCG2_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

SCG2combined = merge(SCG2betas, SCG2_NC, by = "label")
SCG2combined = merge(SCG2combined, target, by = "label")


SCG2betas <- data_summary(SCG2combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(SCG2betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.15), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.01,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

SCG2counts <- data_summary(SCG2combined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(SCG2counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("SCG2-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#------------------------------------------------------------------RELN----
GLMresultsD0vsD20$ResultsPerGene$ENSG00000189056$significantRegulators

#RELN, ENSG00000189056, sig:"cg09949949" "cg04941087" "cg07670722" "cg11912608" "cg23502530" "cg27456945"
RELNbetas = betas2 %>% dplyr::filter(., cgid  %in% c("cg09949949", "cg04941087", "cg07670722", "cg11912608", "cg23502530", "cg27456945"))
RELNbetas = RELNbetas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
RELN_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000189056") 
RELN_NC = RELN_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

RELNcombined = merge(RELNbetas, RELN_NC, by = "label")
RELNcombined = merge(RELNcombined, target, by = "label")

RELNbetas <- data_summary(RELNcombined, varname="Betas", groupnames=c("Day", "cgid"))


a = ggplot(RELNbetas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.40), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.01,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

RELNcounts <- data_summary(RELNcombined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(RELNcounts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("RELN-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#------------------------------------------------------------------DCT----
GLMresultsD0vsD20$ResultsPerGene$ENSG00000080166$significantRegulators

#DCT, ENSG00000080166, sig:"cg24157814" "cg06060467" "cg05256043" "cg05807291" "cg09189889" "cg18293281"
DCTbetas = betas2 %>% dplyr::filter(., cgid  %in% c("cg18910635", "cg05256043", "cg05807291", "cg09189889", "cg18293281"))
DCTbetas = DCTbetas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
DCT_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000080166") 
DCT_NC = DCT_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

DCTcombined = merge(DCTbetas, DCT_NC, by = "label")
DCTcombined = merge(DCTcombined, target, by = "label")

DCTbetas <- data_summary(DCTcombined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(DCTbetas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
   scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.25), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.01,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

DCTcounts <- data_summary(DCTcombined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(DCTcounts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("DCT-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-------------------------------------------------------------GNG11---------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000127920$significantRegulators

#GNG11, ENSG00000127920, sig:"cg00536924" "cg08038054" "cg08236022" "cg25546811"
GNG11betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg00536924", "cg08038054", "cg08236022", "cg25546811"))
GNG11betas = GNG11betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
GNG11_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000127920") 
GNG11_NC = GNG11_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

GNG11combined = merge(GNG11betas, GNG11_NC, by = "label")
GNG11combined = merge(GNG11combined, target, by = "label")

GNG11betas <- data_summary(GNG11combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(GNG11betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.85,0.85), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.01,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

GNG11counts <- data_summary(GNG11combined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(GNG11counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("GNG11-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-------------------------------------------------------------LDHA---------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000134333$significantRegulators

#LDHA, ENSG00000134333, sig:""cg01316516" "cg01381301" "cg02232751" "cg03177631" "cg03628148" "cg07187103" "cg08861826" "cg09436562" "cg11166108" "cg13190306" "cg15700009" "cg19316551" "cg19631472" "cg20429911
LDHAbetas = betas2 %>% dplyr::filter(., cgid  %in% c("cg01316516", "cg01381301", "cg02232751", "cg03177631", "cg03628148", "cg07187103", "cg08861826", "cg09436562", "cg11166108" ,"cg13190306", "cg15700009", "cg19316551", "cg19631472", "cg20429911"))

LDHAbetas = LDHAbetas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
LDHA_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000134333") 
LDHA_NC = LDHA_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

LDHAcombined = merge(LDHAbetas, LDHA_NC, by = "label")
LDHAcombined = merge(LDHAcombined, target, by = "label")

LDHAbetas <- data_summary(LDHAcombined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(LDHAbetas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.85,0.2), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.01,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

LDHAcounts <- data_summary(LDHAcombined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(LDHAcounts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  scale_y_continuous(limits=c(0, 45000)) +
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("LDHA-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-------------------------------------------------------------KRT18---------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000111057$significantRegulators

#KRT18, ENSG00000111057, sig:"cg11722496" "cg16747714" "cg06024289" "cg07051221" "cg07143715" "cg07652628" "cg12121782" "cg14212748" "cg27502493"
KRT18betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg11722496", "cg16747714", "cg06024289", "cg07051221", "cg07143715", "cg07652628", "cg12121782", "cg14212748", "cg27502493"))

KRT18betas = KRT18betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
KRT18_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000111057") 
KRT18_NC = KRT18_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

KRT18combined = merge(KRT18betas, KRT18_NC, by = "label")
KRT18combined = merge(KRT18combined, target, by = "label")

KRT18betas <- data_summary(KRT18combined, varname="Betas", 
                    groupnames=c("Day", "cgid"))

a = ggplot(KRT18betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,
                 position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18, 7, 9, 10, 11, 12, 13, 14))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.85), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.01,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

KRT18counts <- data_summary(KRT18combined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(KRT18counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("KRT18-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-------------------------------------------------------------CDKN1C---------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000129757$significantRegulators

#CDKN1C, ENSG00000129757, sig:"cg00374747" "cg05267118" "cg10929140" "cg11399776" "cg23225147" "cg23904595"
CDKN1Cbetas = betas2 %>% dplyr::filter(., cgid  %in% c("cg00374747", "cg05267118", "cg10929140", "cg11399776", "cg23225147", "cg23904595"))

CDKN1Cbetas = CDKN1Cbetas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
CDKN1C_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000129757") 
CDKN1C_NC = CDKN1C_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

CDKN1Ccombined = merge(CDKN1Cbetas, CDKN1C_NC, by = "label")
CDKN1Ccombined = merge(CDKN1Ccombined, target, by = "label")

CDKN1Cbetas <- data_summary(CDKN1Ccombined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(CDKN1Cbetas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,
                 position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.15), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.01,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

CDKN1Ccounts <- data_summary(CDKN1Ccombined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(CDKN1Ccounts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("CDKN1C-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()

#-------------------------------------------------------------NCAM1---------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000149294$significantRegulators

#NCAM1, ENSG00000149294 , sig:"cg14673319" "cg14971026" "cg00539883" "cg02804231" "cg08743050" "cg12349185" "cg13463897" "cg19624354" "cg20857767"
NCAM1betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg14673319", "cg14971026", "cg00539883" ,"cg02804231", "cg08743050", "cg12349185" ,"cg13463897", "cg19624354" ,"cg20857767"))

NCAM1betas = NCAM1betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
NCAM1_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000149294") 
NCAM1_NC = NCAM1_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

NCAM1combined = merge(NCAM1betas, NCAM1_NC, by = "label")
NCAM1combined = merge(NCAM1combined, target, by = "label")

NCAM1betas <- data_summary(NCAM1combined, varname="Betas", groupnames=c("Day", "cgid"))


a = ggplot(NCAM1betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0, position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.15), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.01,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

NCAM1counts <- data_summary(NCAM1combined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(NCAM1counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("NCAM1-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()
#-------------------------------------------------------------THY1---------
GLMresultsD0vsD20$ResultsPerGene$ENSG00000154096$significantRegulators

#THY1,ENSG00000154096 , sig:"cg14651687" "cg13248115" "cg14850728" "cg15449664" "cg21633698"
THY1betas = betas2 %>% dplyr::filter(., cgid  %in% c("cg14651687", "cg13248115", "cg14850728" ,"cg15449664", "cg21633698"))

THY1betas = THY1betas %>% pivot_longer(cols = -c(cgid), names_to = "label", values_to = "Betas")

normcounts.df.b = normcounts.df[,keep]
normcounts.df.b$Id = rownames(normcounts.df.b)
THY1_NC = normcounts.df.b%>% dplyr::filter(Id == "ENSG00000154096") 
THY1_NC = THY1_NC %>% pivot_longer(cols = -c(Id), names_to = "label", values_to = "NormalizedCounts")

THY1combined = merge(THY1betas, THY1_NC, by = "label")
THY1combined = merge(THY1combined, target, by = "label")

THY1betas <- data_summary(THY1combined, varname="Betas", groupnames=c("Day", "cgid"))

a = ggplot(THY1betas, aes(x = as.factor(Day), y = Betas,group = cgid, col = cgid, shape = cgid))+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=Betas-sd, ymax=Betas+sd), width = 0,position=position_dodge(0.05))+
 scale_color_manual(values = c("#E64B35FF", "#4DBBD5FF", "#00A087FF", "#3C5488FF", "#F39B7FFF", "#8491B4FF", "#91D1C2FF", "#DC0000FF", "#7E6148FF", "#B09C85FF", "#FFCC00", "#996699", "#6666CC", "#FF5722", "#78909C", "#EC407A", "#CE93D8", "#4DD0E1"))+
  scale_shape_manual(values = c(0,17, 1,15, 3,4,19, 5,6, 8, 2, 18))+
  labs(x = "", y = "DNAm levels (beta values)") +
  scale_y_continuous(limits=c(0, 1)) +   theme_few()+
theme(legend.position= c(0.15,0.25), legend.key.size = unit(0.01, "cm"), legend.key.width = unit(0.01,"cm"), legend.text = element_text(size = 4), legend.title = element_text(size = 7))

THY1counts <- data_summary(THY1combined, varname="NormalizedCounts", groupnames=c("Day", "Id"))

b = ggplot(THY1counts, aes(x = as.factor(Day), y = NormalizedCounts,group = Id), col = "Black")+
  geom_line() + 
  geom_point()+
  geom_errorbar(aes(ymin=NormalizedCounts-sd, ymax=NormalizedCounts+sd), width = 0,
                 position=position_dodge(0.05) )+
  theme_few()+
  labs(x = "Day", y = "RNA expression (normalized counts)")

grid.arrange(a, b,
          ncol = 1, nrow = 2)
dev.copy(pdf,paste0("THY1-MORE-individualplots.pdf"),height = 11, width = 6, paper='special')
dev.off()
```

```{r}
sessionInfo()
```